{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aa8dc1fb-404f-4bd0-ae25-571fcc4a7121",
   "metadata": {},
   "source": [
    "## Test read in kraken2 output (without --use-mpa-style)\n",
    "### From the results folder from nf-core https://nf-co.re/mag/results#mag/results-3b4dd3469725654d67e06b3853ba460d64c80788/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "5a4cdcfe-4b7f-46b4-bbec-06e0e310df43",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import altair as alt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "f71d47fd-30b8-4923-b0e1-914511aeba32",
   "metadata": {},
   "outputs": [],
   "source": [
    "kraken_raw= pd.read_csv('/Users/williamrosenbaum/Downloads/kraken2_report.txt', \n",
    "                     sep='\\t',\n",
    "                     header=None,\n",
    "                     names=['pct', 'reads_clade', 'reads_taxon', 'rank', 'ID', 'name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "708b485b-8874-463f-864a-e872e5addd56",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pct</th>\n",
       "      <th>reads_clade</th>\n",
       "      <th>reads_taxon</th>\n",
       "      <th>rank</th>\n",
       "      <th>ID</th>\n",
       "      <th>name</th>\n",
       "      <th>pct_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13879</th>\n",
       "      <td>1.52</td>\n",
       "      <td>214053</td>\n",
       "      <td>214053</td>\n",
       "      <td>S</td>\n",
       "      <td>1211417</td>\n",
       "      <td>uncultured crAssphage</td>\n",
       "      <td>0.994393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14472</th>\n",
       "      <td>0.01</td>\n",
       "      <td>1117</td>\n",
       "      <td>1117</td>\n",
       "      <td>S1</td>\n",
       "      <td>1090134</td>\n",
       "      <td>Salmonella phage SPN3US</td>\n",
       "      <td>0.005189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14522</th>\n",
       "      <td>0.00</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>S</td>\n",
       "      <td>1273755</td>\n",
       "      <td>Halovirus HRTV-8</td>\n",
       "      <td>0.000074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15731</th>\n",
       "      <td>0.00</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>S</td>\n",
       "      <td>1051631</td>\n",
       "      <td>Streptococcus phage YMC-2011</td>\n",
       "      <td>0.000065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17776</th>\n",
       "      <td>0.00</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>S</td>\n",
       "      <td>50294</td>\n",
       "      <td>Psittacid alphaherpesvirus 1</td>\n",
       "      <td>0.000079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17779</th>\n",
       "      <td>0.00</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>S</td>\n",
       "      <td>10317</td>\n",
       "      <td>Cercopithecine alphaherpesvirus 2</td>\n",
       "      <td>0.000065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17936</th>\n",
       "      <td>0.00</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>S</td>\n",
       "      <td>2107708</td>\n",
       "      <td>Pandoravirus neocaledonia</td>\n",
       "      <td>0.000135</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        pct  reads_clade  reads_taxon rank       ID  \\\n",
       "13879  1.52       214053       214053    S  1211417   \n",
       "14472  0.01         1117         1117   S1  1090134   \n",
       "14522  0.00           16           16    S  1273755   \n",
       "15731  0.00           14           14    S  1051631   \n",
       "17776  0.00           17           17    S    50294   \n",
       "17779  0.00           14           14    S    10317   \n",
       "17936  0.00           29           29    S  2107708   \n",
       "\n",
       "                                                name  pct_reads  \n",
       "13879                          uncultured crAssphage   0.994393  \n",
       "14472                        Salmonella phage SPN3US   0.005189  \n",
       "14522                               Halovirus HRTV-8   0.000074  \n",
       "15731                   Streptococcus phage YMC-2011   0.000065  \n",
       "17776                   Psittacid alphaherpesvirus 1   0.000079  \n",
       "17779              Cercopithecine alphaherpesvirus 2   0.000065  \n",
       "17936                      Pandoravirus neocaledonia   0.000135  "
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kraken = (kraken_raw\n",
    "  # Find where the viruses starts in the dataframe \n",
    " .loc[kraken_raw[lambda x: x['name'].str.contains('Viruses')].index[0]:, :]\n",
    " .loc[lambda x: x['reads_taxon'] > 10]\n",
    " .loc[lambda x: x['rank'].str.contains('S')]\n",
    " .assign(pct_reads=lambda x: x['reads_clade'] / x['reads_clade'].sum())\n",
    ")\n",
    "\n",
    "kraken\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "950ad269-e25d-4f74-ada1-06dc533bc399",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-369d5f78251b4116b37df52ad3decbec\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-369d5f78251b4116b37df52ad3decbec\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-369d5f78251b4116b37df52ad3decbec\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm//vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm//vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm//vega-lite@4.17.0?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm//vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"4.17.0\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 400, \"continuousHeight\": 300}}, \"layer\": [{\"mark\": \"bar\", \"encoding\": {\"tooltip\": [{\"field\": \"name\", \"type\": \"nominal\"}, {\"field\": \"reads_taxon\", \"format\": \",\", \"title\": \"Number of reads\", \"type\": \"quantitative\"}], \"x\": {\"field\": \"reads_taxon\", \"type\": \"quantitative\"}, \"y\": {\"field\": \"name\", \"sort\": \"-x\", \"type\": \"nominal\"}}}, {\"mark\": {\"type\": \"text\", \"align\": \"left\", \"dx\": 3}, \"encoding\": {\"text\": {\"field\": \"pct_reads\", \"format\": \".1%\", \"type\": \"quantitative\"}, \"tooltip\": [{\"field\": \"name\", \"type\": \"nominal\"}, {\"field\": \"reads_taxon\", \"format\": \",\", \"title\": \"Number of reads\", \"type\": \"quantitative\"}], \"x\": {\"field\": \"reads_taxon\", \"type\": \"quantitative\"}, \"y\": {\"field\": \"name\", \"sort\": \"-x\", \"type\": \"nominal\"}}}], \"data\": {\"name\": \"data-4b3b7428818443c690b716adde6d2c1d\"}, \"$schema\": \"https://vega.github.io/schema/vega-lite/v4.17.0.json\", \"datasets\": {\"data-4b3b7428818443c690b716adde6d2c1d\": [{\"pct\": 1.52, \"reads_clade\": 214053, \"reads_taxon\": 214053, \"rank\": \"S\", \"ID\": 1211417, \"name\": \"              uncultured crAssphage\", \"pct_reads\": 0.9943928272786398}, {\"pct\": 0.01, \"reads_clade\": 1117, \"reads_taxon\": 1117, \"rank\": \"S1\", \"ID\": 1090134, \"name\": \"            Salmonella phage SPN3US\", \"pct_reads\": 0.00518907367834247}, {\"pct\": 0.0, \"reads_clade\": 16, \"reads_taxon\": 16, \"rank\": \"S\", \"ID\": 1273755, \"name\": \"          Halovirus HRTV-8\", \"pct_reads\": 7.43287187587104e-05}, {\"pct\": 0.0, \"reads_clade\": 14, \"reads_taxon\": 14, \"rank\": \"S\", \"ID\": 1051631, \"name\": \"            Streptococcus phage YMC-2011\", \"pct_reads\": 6.503762891387159e-05}, {\"pct\": 0.0, \"reads_clade\": 17, \"reads_taxon\": 17, \"rank\": \"S\", \"ID\": 50294, \"name\": \"            Psittacid alphaherpesvirus 1\", \"pct_reads\": 7.89742636811298e-05}, {\"pct\": 0.0, \"reads_clade\": 14, \"reads_taxon\": 14, \"rank\": \"S\", \"ID\": 10317, \"name\": \"            Cercopithecine alphaherpesvirus 2\", \"pct_reads\": 6.503762891387159e-05}, {\"pct\": 0.0, \"reads_clade\": 29, \"reads_taxon\": 29, \"rank\": \"S\", \"ID\": 2107708, \"name\": \"            Pandoravirus neocaledonia\", \"pct_reads\": 0.0001347208027501626}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.LayerChart(...)"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bar = alt.Chart(kraken).mark_bar().encode(\n",
    " alt.Y('name:N', sort='-x'),\n",
    " alt.X('reads_taxon:Q'),\n",
    " tooltip=[alt.Tooltip('name:N'),\n",
    "          alt.Tooltip('reads_taxon:Q', title='Number of reads', format=',')]\n",
    ")\n",
    "\n",
    "text = bar.mark_text(\n",
    "    align='left',\n",
    "    dx=3\n",
    ").encode(\n",
    "    alt.Text('pct_reads:Q', format='.1%')\n",
    ")\n",
    "\n",
    "(bar + text)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41812c49-cdf1-4e6e-a30c-f216a5b94e13",
   "metadata": {},
   "source": [
    "## Reading in results from CAT from nf-core\n",
    "### quite some cleaning up "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "b558c788-fa73-4535-9afe-d34c1d398f00",
   "metadata": {},
   "outputs": [],
   "source": [
    "cat = pd.read_csv('MEGAHIT-MaxBin2-CAPES_S11.bin2classification.names.txt',\n",
    "                  sep='\\t', \n",
    "                  skiprows=2,\n",
    "                  header=None,\n",
    "                  usecols=[0, 1, 2, 3, 8, 9, 10, 11, 12, 13, 14, 15],\n",
    "                  names=['bin', 'classification', 'num_ORF',\n",
    "                         'num_ORF_classification_based', 'kingdom', \n",
    "                         'clade', 'clade_phylum', 'phylum_class', 'class_order', \n",
    "                         'order_family', 'family_genus', 'genus'])\n",
    "                 \n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "da54af92-e4ab-4f5b-8672-052d94507a8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin</th>\n",
       "      <th>classification</th>\n",
       "      <th>num_ORF</th>\n",
       "      <th>num_ORF_classification_based</th>\n",
       "      <th>kingdom</th>\n",
       "      <th>clade</th>\n",
       "      <th>clade_phylum</th>\n",
       "      <th>phylum_class</th>\n",
       "      <th>class_order</th>\n",
       "      <th>order_family</th>\n",
       "      <th>family_genus</th>\n",
       "      <th>genus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.002.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>1455</td>\n",
       "      <td>1442</td>\n",
       "      <td>Bacteria (superkingdom): 0.97</td>\n",
       "      <td>FCB group (clade): 0.74</td>\n",
       "      <td>Bacteroidetes/Chlorobi group (clade): 0.74</td>\n",
       "      <td>Bacteroidetes (phylum): 0.74</td>\n",
       "      <td>Bacteroidia (class): 0.74</td>\n",
       "      <td>Bacteroidales (order): 0.74</td>\n",
       "      <td>Bacteroidaceae (family): 0.65</td>\n",
       "      <td>Bacteroides (genus): 0.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.003.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>1790</td>\n",
       "      <td>1782</td>\n",
       "      <td>Bacteria (superkingdom): 0.97</td>\n",
       "      <td>Terrabacteria group (clade): 0.82</td>\n",
       "      <td>Actinobacteria (phylum): 0.71</td>\n",
       "      <td>Actinobacteria (class): 0.71</td>\n",
       "      <td>Bifidobacteriales (order): 0.71</td>\n",
       "      <td>Bifidobacteriaceae (family): 0.71</td>\n",
       "      <td>Bifidobacterium (genus): 0.71</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.004.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2022</td>\n",
       "      <td>2021</td>\n",
       "      <td>Bacteria (superkingdom): 0.72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.005.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>6074</td>\n",
       "      <td>6024</td>\n",
       "      <td>Bacteria (superkingdom): 0.98</td>\n",
       "      <td>FCB group (clade): 0.46</td>\n",
       "      <td>Bacteroidetes/Chlorobi group (clade): 0.46</td>\n",
       "      <td>Bacteroidetes (phylum): 0.46</td>\n",
       "      <td>Bacteroidia (class): 0.46</td>\n",
       "      <td>Bacteroidales (order): 0.46</td>\n",
       "      <td>Bacteroidaceae (family): 0.40</td>\n",
       "      <td>Bacteroides (genus): 0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.006.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2721</td>\n",
       "      <td>2717</td>\n",
       "      <td>Bacteria (superkingdom): 0.83</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                bin classification  num_ORF  \\\n",
       "0  MEGAHIT-MaxBin2-CAPES_S11.002.fa     classified     1455   \n",
       "1  MEGAHIT-MaxBin2-CAPES_S11.003.fa     classified     1790   \n",
       "2  MEGAHIT-MaxBin2-CAPES_S11.004.fa     classified     2022   \n",
       "3  MEGAHIT-MaxBin2-CAPES_S11.005.fa     classified     6074   \n",
       "4  MEGAHIT-MaxBin2-CAPES_S11.006.fa     classified     2721   \n",
       "\n",
       "   num_ORF_classification_based                        kingdom  \\\n",
       "0                          1442  Bacteria (superkingdom): 0.97   \n",
       "1                          1782  Bacteria (superkingdom): 0.97   \n",
       "2                          2021  Bacteria (superkingdom): 0.72   \n",
       "3                          6024  Bacteria (superkingdom): 0.98   \n",
       "4                          2717  Bacteria (superkingdom): 0.83   \n",
       "\n",
       "                               clade  \\\n",
       "0            FCB group (clade): 0.74   \n",
       "1  Terrabacteria group (clade): 0.82   \n",
       "2                                NaN   \n",
       "3            FCB group (clade): 0.46   \n",
       "4                                NaN   \n",
       "\n",
       "                                 clade_phylum                  phylum_class  \\\n",
       "0  Bacteroidetes/Chlorobi group (clade): 0.74  Bacteroidetes (phylum): 0.74   \n",
       "1               Actinobacteria (phylum): 0.71  Actinobacteria (class): 0.71   \n",
       "2                                         NaN                           NaN   \n",
       "3  Bacteroidetes/Chlorobi group (clade): 0.46  Bacteroidetes (phylum): 0.46   \n",
       "4                                         NaN                           NaN   \n",
       "\n",
       "                       class_order                       order_family  \\\n",
       "0        Bacteroidia (class): 0.74        Bacteroidales (order): 0.74   \n",
       "1  Bifidobacteriales (order): 0.71  Bifidobacteriaceae (family): 0.71   \n",
       "2                              NaN                                NaN   \n",
       "3        Bacteroidia (class): 0.46        Bacteroidales (order): 0.46   \n",
       "4                              NaN                                NaN   \n",
       "\n",
       "                    family_genus                      genus  \n",
       "0  Bacteroidaceae (family): 0.65  Bacteroides (genus): 0.65  \n",
       "1  Bifidobacterium (genus): 0.71                        NaN  \n",
       "2                            NaN                        NaN  \n",
       "3  Bacteroidaceae (family): 0.40  Bacteroides (genus): 0.40  \n",
       "4                            NaN                        NaN  "
      ]
     },
     "execution_count": 289,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cat.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "id": "e5d9191d-3072-4efa-af5e-a43842be2541",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin</th>\n",
       "      <th>classification</th>\n",
       "      <th>num_ORF</th>\n",
       "      <th>num_ORF_classification_based</th>\n",
       "      <th>value</th>\n",
       "      <th>certainty</th>\n",
       "      <th>name</th>\n",
       "      <th>order</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.002.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>1455</td>\n",
       "      <td>1442</td>\n",
       "      <td>Bacteria (superkingdom): 0.97</td>\n",
       "      <td>0.97</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.003.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>1790</td>\n",
       "      <td>1782</td>\n",
       "      <td>Bacteria (superkingdom): 0.97</td>\n",
       "      <td>0.97</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.004.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2022</td>\n",
       "      <td>2021</td>\n",
       "      <td>Bacteria (superkingdom): 0.72</td>\n",
       "      <td>0.72</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.005.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>6074</td>\n",
       "      <td>6024</td>\n",
       "      <td>Bacteria (superkingdom): 0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.006.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2721</td>\n",
       "      <td>2717</td>\n",
       "      <td>Bacteria (superkingdom): 0.83</td>\n",
       "      <td>0.83</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>superkingdom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.009.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>4822</td>\n",
       "      <td>4781</td>\n",
       "      <td>Bacteroides (genus): 0.69</td>\n",
       "      <td>0.69</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>genus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.016.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2318</td>\n",
       "      <td>2280</td>\n",
       "      <td>Acidaminococcus massiliensis (species): 0.49</td>\n",
       "      <td>0.49</td>\n",
       "      <td>Acidaminococcus massiliensis</td>\n",
       "      <td>species</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.022.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2097</td>\n",
       "      <td>1994</td>\n",
       "      <td>unclassified Olsenella (no rank): 0.82</td>\n",
       "      <td>0.82</td>\n",
       "      <td>unclassified Olsenella</td>\n",
       "      <td>no rank</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.028.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>6336</td>\n",
       "      <td>6098</td>\n",
       "      <td>Bacteroides (genus): 0.59</td>\n",
       "      <td>0.59</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>genus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S11.030.fa</td>\n",
       "      <td>classified</td>\n",
       "      <td>2367</td>\n",
       "      <td>2312</td>\n",
       "      <td>Alistipes (genus): 0.62</td>\n",
       "      <td>0.62</td>\n",
       "      <td>Alistipes</td>\n",
       "      <td>genus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>154 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  bin classification  num_ORF  \\\n",
       "0    MEGAHIT-MaxBin2-CAPES_S11.002.fa     classified     1455   \n",
       "1    MEGAHIT-MaxBin2-CAPES_S11.003.fa     classified     1790   \n",
       "2    MEGAHIT-MaxBin2-CAPES_S11.004.fa     classified     2022   \n",
       "3    MEGAHIT-MaxBin2-CAPES_S11.005.fa     classified     6074   \n",
       "4    MEGAHIT-MaxBin2-CAPES_S11.006.fa     classified     2721   \n",
       "..                                ...            ...      ...   \n",
       "231  MEGAHIT-MaxBin2-CAPES_S11.009.fa     classified     4822   \n",
       "238  MEGAHIT-MaxBin2-CAPES_S11.016.fa     classified     2318   \n",
       "244  MEGAHIT-MaxBin2-CAPES_S11.022.fa     classified     2097   \n",
       "250  MEGAHIT-MaxBin2-CAPES_S11.028.fa     classified     6336   \n",
       "252  MEGAHIT-MaxBin2-CAPES_S11.030.fa     classified     2367   \n",
       "\n",
       "     num_ORF_classification_based  \\\n",
       "0                            1442   \n",
       "1                            1782   \n",
       "2                            2021   \n",
       "3                            6024   \n",
       "4                            2717   \n",
       "..                            ...   \n",
       "231                          4781   \n",
       "238                          2280   \n",
       "244                          1994   \n",
       "250                          6098   \n",
       "252                          2312   \n",
       "\n",
       "                                            value  certainty  \\\n",
       "0                   Bacteria (superkingdom): 0.97       0.97   \n",
       "1                   Bacteria (superkingdom): 0.97       0.97   \n",
       "2                   Bacteria (superkingdom): 0.72       0.72   \n",
       "3                   Bacteria (superkingdom): 0.98       0.98   \n",
       "4                   Bacteria (superkingdom): 0.83       0.83   \n",
       "..                                            ...        ...   \n",
       "231                     Bacteroides (genus): 0.69       0.69   \n",
       "238  Acidaminococcus massiliensis (species): 0.49       0.49   \n",
       "244        unclassified Olsenella (no rank): 0.82       0.82   \n",
       "250                     Bacteroides (genus): 0.59       0.59   \n",
       "252                       Alistipes (genus): 0.62       0.62   \n",
       "\n",
       "                              name         order  \n",
       "0                        Bacteria   superkingdom  \n",
       "1                        Bacteria   superkingdom  \n",
       "2                        Bacteria   superkingdom  \n",
       "3                        Bacteria   superkingdom  \n",
       "4                        Bacteria   superkingdom  \n",
       "..                             ...           ...  \n",
       "231                   Bacteroides          genus  \n",
       "238  Acidaminococcus massiliensis        species  \n",
       "244        unclassified Olsenella        no rank  \n",
       "250                   Bacteroides          genus  \n",
       "252                     Alistipes          genus  \n",
       "\n",
       "[154 rows x 8 columns]"
      ]
     },
     "execution_count": 385,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(cat\n",
    " .melt(id_vars=['bin', 'classification', 'num_ORF', \n",
    "                'num_ORF_classification_based'])\n",
    " .loc[lambda x: ~x['value'].isna()]\n",
    " .drop(columns='variable')\n",
    " #.value_counts('bin')\n",
    " #.loc[lambda x: x['bin'] == 'MEGAHIT-MaxBin2-CAPES_S11.002.fa']\n",
    " .assign(certainty=lambda x: x['value'].str.extract(r'(\\d+.*)').astype(float),\n",
    "         name=lambda x: x['value'].str.extract(r'(.*?)\\('),\n",
    "         order=lambda x: x['value'].str.extract(r'.*\\((.*?)\\)'))\n",
    " .loc[lambda x: x['order'] != 'clade']\n",
    " \n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c7adc93-ea33-452d-a14b-05d407ca28e4",
   "metadata": {},
   "source": [
    "### CAT LCA file\n",
    "\n",
    "### maybe the lca file is nothing to have? seems to be redundant if you have the bin2classificaiotn file. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd90b680-0062-4047-872b-0374b1eb4755",
   "metadata": {},
   "source": [
    "## Reading in results from gtdb-tk from nf-core (the summary tsv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 492,
   "id": "a7222955-007c-4940-99ae-d17d2f9ee550",
   "metadata": {},
   "outputs": [],
   "source": [
    "gtdb_raw = pd.read_csv('gtdbtk_summary.tsv',\n",
    "                  sep='\\t')\n",
    "\n",
    "def massage_gtdb(gtwb_raw: pd.DataFrame, \n",
    "                 order: str, \n",
    "                 sample_name: str, \n",
    "                 assembly_type: str) -> pd.DataFrame:\n",
    "    \n",
    "    rank = {'order': 'o__', 'family': 'f__'}\n",
    "    \n",
    "    return (gtdb_raw\n",
    "     .loc[lambda x: x['user_genome'].str.contains(f'(?=.*{sample_name})(?=.*{assembly_type})')]\n",
    "     .loc[lambda x: ~x['classification'].isna()]\n",
    "     .assign(classification=lambda x: x['classification'].str.split(';'))\n",
    "     .explode(column='classification')\n",
    "     .loc[lambda x: x['classification'].str.contains(rank[order])]\n",
    "     .assign(classification=lambda x: x['classification'].str.replace(rank[order], ''))\n",
    "    )\n",
    "    \n",
    "gtdb = massage_gtdb(gtdb_raw, 'order', '_S11', 'MEGA')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 489,
   "id": "9f9ea98f-f643-47df-aab3-d84c988c8bfa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-31ec5ad24adf45958ccad271a2d6a2f7\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-31ec5ad24adf45958ccad271a2d6a2f7\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-31ec5ad24adf45958ccad271a2d6a2f7\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm//vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm//vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm//vega-lite@4.17.0?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm//vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"4.17.0\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 400, \"continuousHeight\": 300}}, \"data\": {\"name\": \"data-813f6c0b25f3adccc4b23fd2a079e90c\"}, \"mark\": \"bar\", \"encoding\": {\"x\": {\"aggregate\": \"count\", \"field\": \"classification\", \"type\": \"quantitative\"}, \"y\": {\"field\": \"classification\", \"sort\": \"-x\", \"type\": \"nominal\"}}, \"$schema\": \"https://vega.github.io/schema/vega-lite/v4.17.0.json\", \"datasets\": {\"data-813f6c0b25f3adccc4b23fd2a079e90c\": [{\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.003.fa\", \"classification\": \"Actinomycetales\", \"fastani_reference\": \"GCF_000196555.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium longum\", \"fastani_ani\": 98.84, \"fastani_af\": 0.94, \"closest_placement_reference\": \"GCF_000196555.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium longum\", \"closest_placement_ani\": 98.84, \"closest_placement_af\": 0.94, \"pplacer_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000269965.1, s__Bifidobacterium infantis, 95.0, 95.19, 0.86; GCF_001025175.1, s__Bifidobacterium breve, 95.0, 86.61, 0.76; GCF_003129905.1, s__Bifidobacterium callitrichidarum, 95.0, 83.93, 0.77; GCF_002259745.1, s__Bifidobacterium myosotis, 95.0, 83.27, 0.73; GCF_000741695.1, s__Bifidobacterium reuteri, 95.0, 83.13, 0.65; GCF_009193305.1, s__Bifidobacterium sp009193305, 95.0, 82.89, 0.65; GCF_002802915.1, s__Bifidobacterium felsineum, 95.0, 82.77, 0.65; GCF_008698235.1, s__Bifidobacterium rousetti, 95.0, 82.46, 0.58; GCF_012932375.1, s__Bifidobacterium sp012932375, 95.0, 82.45, 0.64; GCF_000741715.1, s__Bifidobacterium saguini, 95.0, 82.28, 0.63; GCF_002860405.1, s__Bifidobacterium imperatoris, 95.0, 82.24, 0.63; GCF_002860365.1, s__Bifidobacterium parmae, 95.0, 82.24, 0.58; GCF_002802865.1, s__Bifidobacterium scaligerum, 95.0, 82.23, 0.64; GCF_010667645.1, s__Bifidobacterium sp010667645, 95.0, 82.22, 0.6; GCF_000741785.1, s__Bifidobacterium stellenboschense, 95.0, 82.04, 0.6; GCF_001042635.1, s__Bifidobacterium scardovii, 95.0, 82.02, 0.64; GCF_000741175.1, s__Bifidobacterium callitrichos, 95.0, 81.97, 0.62; GCF_001417815.1, s__Bifidobacterium aesculapii, 95.0, 81.96, 0.57; GCF_001025135.1, s__Bifidobacterium bifidum, 95.0, 81.67, 0.51; GCF_012932365.1, s__Bifidobacterium sp012932365, 95.0, 81.6, 0.5; GCF_009299475.1, s__Bifidobacterium avesanii, 95.0, 81.52, 0.55; GCF_000010425.1, s__Bifidobacterium adolescentis, 95.0, 81.35, 0.51; GCF_008698145.1, s__Bifidobacterium vespertilionis, 95.0, 81.32, 0.51; GCF_002259685.1, s__Bifidobacterium eulemuris, 95.0, 81.31, 0.56; GCF_012932425.1, s__Bifidobacterium sp012932425, 95.0, 81.3, 0.61; GCF_012932675.1, s__Bifidobacterium sp012932675, 95.0, 81.22, 0.52; GCF_010667685.1, s__Bifidobacterium aerophilum, 95.0, 81.21, 0.57; GCF_001895165.1, s__Bifidobacterium lemurum, 95.0, 81.19, 0.58; GCF_009193355.1, s__Bifidobacterium sp009193355, 95.0, 81.16, 0.53; GCF_000741165.1, s__Bifidobacterium biavatii, 95.0, 81.14, 0.56; GCF_009299505.1, s__Bifidobacterium ramosum, 95.0, 81.12, 0.59; GCF_900129045.1, s__Bifidobacterium merycicum, 95.0, 81.03, 0.5; GCF_000770925.1, s__Bifidobacterium ruminantium, 95.0, 80.97, 0.46; GCF_001025155.1, s__Bifidobacterium angulatum, 95.0, 80.97, 0.44; GCF_001042615.1, s__Bifidobacterium kashiwanohense, 96.3948, 80.73, 0.45; GCF_003952945.1, s__Bifidobacterium samirii, 95.0, 80.72, 0.54; GCA_002451435.1, s__Bifidobacterium sp002451435, 95.0, 80.7, 0.45; GCF_000741215.1, s__Bifidobacterium gallinarum, 95.0, 80.63, 0.5; GCF_000771405.1, s__Bifidobacterium pullorum, 95.0, 80.6, 0.5; GCF_009078285.1, s__Bifidobacterium jacchi, 95.0, 80.59, 0.49; GCF_001025215.1, s__Bifidobacterium pseudocatenulatum, 95.0, 80.3, 0.46; GCF_001025195.1, s__Bifidobacterium catenulatum, 96.3948, 80.27, 0.4; GCF_002742445.1, s__Bifidobacterium sp002742445, 95.0, 80.2, 0.43; GCF_001042595.1, s__Bifidobacterium dentium, 95.0, 80.19, 0.45; GCF_000800455.1, s__Bifidobacterium kashiwanohense_A, 95.1942, 80.13, 0.46; GCF_003951095.1, s__Bifidobacterium goeldii, 95.0, 79.98, 0.35; GCF_000522505.1, s__Bifidobacterium moukalabense, 95.0, 79.83, 0.51; GCF_012932445.1, s__Bifidobacterium sp012932445, 95.0, 79.78, 0.35; GCA_002298605.1, s__Bifidobacterium sp002298605, 95.0, 79.77, 0.37; GCF_010667615.1, s__Bifidobacterium sp010667615, 95.0, 79.76, 0.37; GCF_000741575.1, s__Bifidobacterium cuniculi, 95.0, 79.73, 0.4; GCF_000741295.1, s__Bifidobacterium globosum, 95.0, 79.37, 0.36; GCF_004155535.1, s__Bifidobacterium pseudolongum_C, 95.0, 79.3, 0.38; GCF_000771265.1, s__Bifidobacterium thermophilum, 95.0, 79.25, 0.36; GCF_000741135.1, s__Bifidobacterium choerinum, 95.0, 79.16, 0.38; GCA_000741495.1, s__Bifidobacterium thermophilum_A, 95.0, 79.13, 0.35; GCF_002860345.1, s__Bifidobacterium anseris, 95.0, 79.03, 0.35; GCF_000741535.1, s__Bifidobacterium boum, 95.0, 79.03, 0.33; GCF_000771225.1, s__Bifidobacterium pseudolongum, 95.0, 79.02, 0.35; GCF_003952025.1, s__Bifidobacterium castoris, 95.0, 78.99, 0.38; GCF_002286915.1, s__Bifidobacterium italicum, 95.0, 78.96, 0.36; GCF_002259755.1, s__Bifidobacterium hapali, 95.0, 78.89, 0.29; GCF_009299485.1, s__Bifidobacterium apri, 95.0, 78.88, 0.32; GCF_000741255.1, s__Bifidobacterium magnum, 95.0, 78.79, 0.28; GCF_000741775.1, s__Bifidobacterium subtile, 95.0, 78.68, 0.39; GCF_009371885.1, s__Bifidobacterium tibiigranuli, 95.0, 78.57, 0.36; GCF_000741285.1, s__Bifidobacterium mongoliense, 95.0, 78.49, 0.32; GCF_000260715.1, s__Bifidobacterium animalis, 95.0, 78.39, 0.34; GCF_001263395.1, s__Bifidobacterium actinocoloniiforme, 95.0, 78.14, 0.19; GCF_009727065.1, s__Bifidobacterium sp900551485, 95.0, 78.14, 0.33; GCF_002286935.1, s__Bifidobacterium criceti, 95.0, 78.08, 0.31; GCF_003315635.1, s__Bifidobacterium aemilianum, 95.0, 78.02, 0.27; GCF_000741205.1, s__Bifidobacterium gallicum, 95.0, 78.01, 0.23; GCF_000741645.1, s__Bifidobacterium minimum, 95.0, 77.85, 0.22; GCF_000706765.1, s__Bifidobacterium indicum, 95.0, 77.82, 0.17; GCF_000263635.1, s__Bifidobacterium vaginale_C, 95.0, 77.76, 0.05; GCF_003202755.1, s__Bifidobacterium asteroides_F, 95.0, 77.76, 0.19; GCF_000741525.1, s__Bifidobacterium bohemicum, 95.0, 77.73, 0.23; GCF_003202715.1, s__Bifidobacterium asteroides_E, 95.0, 77.71, 0.19; GCF_000741765.1, s__Bifidobacterium tsurumiense, 95.0, 77.56, 0.19; GCF_001546485.1, s__Bifidobacterium vaginale_D, 95.0, 77.53, 0.05; GCF_000263595.1, s__Bifidobacterium vaginale_E, 95.0, 77.52, 0.05; GCF_003951975.1, s__Bifidobacterium dolichotidis, 95.0, 77.48, 0.14; GCF_003408845.1, s__Bifidobacterium vaginale_H, 95.0, 77.47, 0.05; GCF_003315615.1, s__Bifidobacterium xylocopae, 95.0, 77.46, 0.19; GCF_000738005.1, s__Bifidobacterium crudilactis, 95.0, 77.44, 0.19; GCF_001563665.1, s__Bifidobacterium vaginale_A, 95.0, 77.43, 0.03; GCF_003585845.1, s__Bifidobacterium sp003585845, 95.0, 77.42, 0.05; GCF_900094885.1, s__Bifidobacterium commune, 95.0, 77.33, 0.14; GCF_000741705.1, s__Bifidobacterium psychraerophilum, 95.0, 77.32, 0.21; GCF_002896555.1, s__Bifidobacterium vaginale_F, 95.0, 77.16, 0.05; GCF_003202695.1, s__Bifidobacterium asteroides_G, 95.0, 77.12, 0.17; GCF_003585735.1, s__Bifidobacterium sp003585735, 95.0, 77.08, 0.06; GCF_000025205.1, s__Bifidobacterium vaginale_G, 95.0, 76.98, 0.05; GCF_001546455.1, s__Bifidobacterium vaginale_B, 95.0, 76.86, 0.06; GCF_001042655.1, s__Bifidobacterium vaginale, 95.0, 76.81, 0.05; GCF_000737845.1, s__Bifidobacterium bombi, 95.0, 76.71, 0.13; GCF_002884815.1, s__Bifidobacterium sp002884815, 95.0, 76.69, 0.05; GCF_002259795.1, s__Bifidobacterium aquikefiri, 95.0, 76.56, 0.08\", \"msa_percent\": 88.8, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.004.fa\", \"classification\": \"Enterobacterales\", \"fastani_reference\": \"GCF_002950215.1\", \"fastani_reference_radius\": 97.0028, \"fastani_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia;s__Escherichia flexneri\", \"fastani_ani\": 98.03, \"fastani_af\": 0.85, \"closest_placement_reference\": null, \"closest_placement_radius\": null, \"closest_placement_taxonomy\": null, \"closest_placement_ani\": null, \"closest_placement_af\": null, \"pplacer_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia;s__\", \"classification_method\": \"ANI\", \"note\": \"topological placement and ANI have incongruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_002950215.1, s__Escherichia flexneri, 97.0028, 98.03, 0.85; GCF_000026325.1, s__Escherichia coli_D, 96.9551, 97.59, 0.88; GCF_002949675.1, s__Escherichia dysenteriae, 97.0028, 97.41, 0.8; GCF_003697165.2, s__Escherichia coli, 96.8144, 97.21, 0.86; GCA_003018335.1, s__Escherichia coli_C, 96.1333, 96.14, 0.87; GCF_002965065.1, s__Escherichia sp002965065, 95.0, 94.76, 0.75; GCF_004211955.1, s__Escherichia sp004211955, 95.0, 93.9, 0.84; GCF_005843885.1, s__Escherichia sp005843885, 95.0, 93.56, 0.86; GCF_000208585.1, s__Escherichia sp000208585, 95.0, 93.51, 0.86; GCF_011881725.1, s__Escherichia coli_E, 95.0, 93.03, 0.82; GCF_001660175.1, s__Escherichia sp001660175, 95.0, 92.67, 0.85; GCF_000026225.1, s__Escherichia fergusonii, 95.0, 92.11, 0.76; GCF_002900365.1, s__Escherichia marmotae, 95.0, 91.81, 0.84; GCF_000759775.1, s__Escherichia albertii, 95.0, 91.31, 0.75\", \"msa_percent\": 63.65, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.008.fa\", \"classification\": \"Veillonellales\", \"fastani_reference\": \"GCF_000417505.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Megasphaera;s__Megasphaera sp000417505\", \"fastani_ani\": 99.34, \"fastani_af\": 0.85, \"closest_placement_reference\": \"GCF_000417505.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Megasphaera;s__Megasphaera sp000417505\", \"closest_placement_ani\": 99.34, \"closest_placement_af\": 0.85, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Megasphaera;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_902388315.1, s__Megasphaera sp902388315, 95.0, 91.88, 0.77; GCF_003010495.1, s__Megasphaera elsdenii, 95.0, 90.96, 0.82; GCA_900548635.1, s__Megasphaera sp900548635, 95.0, 87.33, 0.87; GCF_001546855.1, s__Megasphaera sp001546855, 95.0, 82.97, 0.67; GCA_900539295.1, s__Megasphaera sp900539295, 95.0, 82.93, 0.65; GCF_000455225.1, s__Megasphaera massiliensis, 95.0, 82.51, 0.66; GCA_002319965.1, s__Megasphaera sp002319965, 95.0, 82.26, 0.63; GCF_003367905.1, s__Megasphaera stantonii, 95.0, 79.73, 0.47; GCA_900554215.1, s__Megasphaera sp900554215, 95.0, 79.22, 0.43; GCA_002431345.1, s__Megasphaera sp002431345, 95.0, 78.27, 0.26; GCA_900540735.1, s__Megasphaera sp900540735, 95.0, 77.94, 0.28; GCF_001045675.1, s__Megasphaera cerevisiae, 95.0, 77.78, 0.16\", \"msa_percent\": 83.52, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.010.fa\", \"classification\": \"Actinomycetales\", \"fastani_reference\": \"GCF_000010425.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium adolescentis\", \"fastani_ani\": 98.23, \"fastani_af\": 0.84, \"closest_placement_reference\": \"GCF_000010425.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium adolescentis\", \"closest_placement_ani\": 98.23, \"closest_placement_af\": 0.84, \"pplacer_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000770925.1, s__Bifidobacterium ruminantium, 95.0, 88.64, 0.74; GCF_001025195.1, s__Bifidobacterium catenulatum, 96.3948, 84.97, 0.69; GCF_001042615.1, s__Bifidobacterium kashiwanohense, 96.3948, 84.55, 0.71; GCF_001025215.1, s__Bifidobacterium pseudocatenulatum, 95.0, 84.41, 0.69; GCF_002742445.1, s__Bifidobacterium sp002742445, 95.0, 84.28, 0.67; GCA_002451435.1, s__Bifidobacterium sp002451435, 95.0, 84.12, 0.63; GCF_000800455.1, s__Bifidobacterium kashiwanohense_A, 95.1942, 84.1, 0.67; GCF_000522505.1, s__Bifidobacterium moukalabense, 95.0, 83.55, 0.68; GCF_001042595.1, s__Bifidobacterium dentium, 95.0, 83.11, 0.66; GCF_012932675.1, s__Bifidobacterium sp012932675, 95.0, 82.99, 0.57; GCF_001025155.1, s__Bifidobacterium angulatum, 95.0, 82.79, 0.54; GCF_900129045.1, s__Bifidobacterium merycicum, 95.0, 82.78, 0.57; GCF_000269965.1, s__Bifidobacterium infantis, 95.0, 81.58, 0.51; GCF_000196555.1, s__Bifidobacterium longum, 95.0, 81.43, 0.49; GCF_002860365.1, s__Bifidobacterium parmae, 95.0, 81.36, 0.52; GCF_003129905.1, s__Bifidobacterium callitrichidarum, 95.0, 81.35, 0.51; GCF_010667645.1, s__Bifidobacterium sp010667645, 95.0, 81.29, 0.54; GCF_000741785.1, s__Bifidobacterium stellenboschense, 95.0, 81.16, 0.52; GCF_010667685.1, s__Bifidobacterium aerophilum, 95.0, 81.15, 0.54; GCF_001895165.1, s__Bifidobacterium lemurum, 95.0, 81.07, 0.52; GCF_009299475.1, s__Bifidobacterium avesanii, 95.0, 81.04, 0.47; GCF_001042635.1, s__Bifidobacterium scardovii, 95.0, 81.01, 0.55; GCF_009299505.1, s__Bifidobacterium ramosum, 95.0, 81.0, 0.56; GCF_000741165.1, s__Bifidobacterium biavatii, 95.0, 80.98, 0.53; GCF_002259685.1, s__Bifidobacterium eulemuris, 95.0, 80.96, 0.52; GCF_003952945.1, s__Bifidobacterium samirii, 95.0, 80.96, 0.51; GCF_000741175.1, s__Bifidobacterium callitrichos, 95.0, 80.94, 0.52; GCF_001025135.1, s__Bifidobacterium bifidum, 95.0, 80.9, 0.48; GCF_001025175.1, s__Bifidobacterium breve, 95.0, 80.8, 0.45; GCF_009193355.1, s__Bifidobacterium sp009193355, 95.0, 80.78, 0.5; GCF_000771405.1, s__Bifidobacterium pullorum, 95.0, 80.75, 0.46; GCF_009193305.1, s__Bifidobacterium sp009193305, 95.0, 80.74, 0.45; GCF_008698235.1, s__Bifidobacterium rousetti, 95.0, 80.72, 0.52; GCF_012932365.1, s__Bifidobacterium sp012932365, 95.0, 80.7, 0.48; GCF_002259745.1, s__Bifidobacterium myosotis, 95.0, 80.69, 0.52; GCF_000741215.1, s__Bifidobacterium gallinarum, 95.0, 80.68, 0.45; GCF_001417815.1, s__Bifidobacterium aesculapii, 95.0, 80.65, 0.51; GCF_008698145.1, s__Bifidobacterium vespertilionis, 95.0, 80.64, 0.45; GCF_002860405.1, s__Bifidobacterium imperatoris, 95.0, 80.62, 0.39; GCF_000741695.1, s__Bifidobacterium reuteri, 95.0, 80.62, 0.48; GCF_002802915.1, s__Bifidobacterium felsineum, 95.0, 80.57, 0.43; GCF_000741715.1, s__Bifidobacterium saguini, 95.0, 80.37, 0.41; GCF_012932375.1, s__Bifidobacterium sp012932375, 95.0, 80.31, 0.44; GCF_002802865.1, s__Bifidobacterium scaligerum, 95.0, 80.28, 0.44; GCF_009078285.1, s__Bifidobacterium jacchi, 95.0, 80.17, 0.43; GCF_012932425.1, s__Bifidobacterium sp012932425, 95.0, 80.13, 0.45; GCF_000741575.1, s__Bifidobacterium cuniculi, 95.0, 80.08, 0.39; GCA_002298605.1, s__Bifidobacterium sp002298605, 95.0, 79.96, 0.39; GCF_012932445.1, s__Bifidobacterium sp012932445, 95.0, 79.95, 0.34; GCF_003951095.1, s__Bifidobacterium goeldii, 95.0, 79.9, 0.33; GCF_010667615.1, s__Bifidobacterium sp010667615, 95.0, 79.85, 0.4; GCF_000741135.1, s__Bifidobacterium choerinum, 95.0, 79.77, 0.42; GCF_000741295.1, s__Bifidobacterium globosum, 95.0, 79.69, 0.38; GCF_000741255.1, s__Bifidobacterium magnum, 95.0, 79.67, 0.32; GCF_000741535.1, s__Bifidobacterium boum, 95.0, 79.67, 0.36; GCA_000741495.1, s__Bifidobacterium thermophilum_A, 95.0, 79.66, 0.4; GCF_000771265.1, s__Bifidobacterium thermophilum, 95.0, 79.64, 0.38; GCF_000771225.1, s__Bifidobacterium pseudolongum, 95.0, 79.63, 0.37; GCF_004155535.1, s__Bifidobacterium pseudolongum_C, 95.0, 79.62, 0.4; GCF_002286915.1, s__Bifidobacterium italicum, 95.0, 79.56, 0.4; GCF_003952025.1, s__Bifidobacterium castoris, 95.0, 79.56, 0.41; GCF_009299485.1, s__Bifidobacterium apri, 95.0, 79.49, 0.37; GCF_002860345.1, s__Bifidobacterium anseris, 95.0, 79.34, 0.41; GCF_000260715.1, s__Bifidobacterium animalis, 95.0, 79.22, 0.36; GCF_009727065.1, s__Bifidobacterium sp900551485, 95.0, 79.06, 0.34; GCF_000741285.1, s__Bifidobacterium mongoliense, 95.0, 78.92, 0.29; GCF_002259755.1, s__Bifidobacterium hapali, 95.0, 78.88, 0.29; GCF_009371885.1, s__Bifidobacterium tibiigranuli, 95.0, 78.88, 0.37; GCF_000741775.1, s__Bifidobacterium subtile, 95.0, 78.8, 0.38; GCF_002286935.1, s__Bifidobacterium criceti, 95.0, 78.75, 0.34; GCF_000741205.1, s__Bifidobacterium gallicum, 95.0, 78.5, 0.25; GCF_000741765.1, s__Bifidobacterium tsurumiense, 95.0, 78.39, 0.18; GCF_003315635.1, s__Bifidobacterium aemilianum, 95.0, 78.36, 0.23; GCF_000741525.1, s__Bifidobacterium bohemicum, 95.0, 78.07, 0.2; GCF_000741645.1, s__Bifidobacterium minimum, 95.0, 78.05, 0.23; GCF_003951975.1, s__Bifidobacterium dolichotidis, 95.0, 78.02, 0.13; GCF_003315615.1, s__Bifidobacterium xylocopae, 95.0, 77.95, 0.18; GCF_003202755.1, s__Bifidobacterium asteroides_F, 95.0, 77.91, 0.16; GCF_000706765.1, s__Bifidobacterium indicum, 95.0, 77.88, 0.15; GCF_900094885.1, s__Bifidobacterium commune, 95.0, 77.86, 0.17; GCF_001263395.1, s__Bifidobacterium actinocoloniiforme, 95.0, 77.85, 0.18; GCF_003408845.1, s__Bifidobacterium vaginale_H, 95.0, 77.84, 0.06; GCF_001042655.1, s__Bifidobacterium vaginale, 95.0, 77.82, 0.05; GCF_003585845.1, s__Bifidobacterium sp003585845, 95.0, 77.78, 0.06; GCF_003202715.1, s__Bifidobacterium asteroides_E, 95.0, 77.77, 0.15; GCF_000738005.1, s__Bifidobacterium crudilactis, 95.0, 77.68, 0.19; GCF_003202695.1, s__Bifidobacterium asteroides_G, 95.0, 77.62, 0.14; GCF_000263635.1, s__Bifidobacterium vaginale_C, 95.0, 77.6, 0.06; GCF_001546485.1, s__Bifidobacterium vaginale_D, 95.0, 77.55, 0.06; GCF_002896555.1, s__Bifidobacterium vaginale_F, 95.0, 77.43, 0.05; GCF_003585735.1, s__Bifidobacterium sp003585735, 95.0, 77.35, 0.09; GCF_000263595.1, s__Bifidobacterium vaginale_E, 95.0, 77.29, 0.06; GCF_001563665.1, s__Bifidobacterium vaginale_A, 95.0, 77.24, 0.04; GCF_000741705.1, s__Bifidobacterium psychraerophilum, 95.0, 77.15, 0.21; GCF_000025205.1, s__Bifidobacterium vaginale_G, 95.0, 77.14, 0.07; GCF_001546455.1, s__Bifidobacterium vaginale_B, 95.0, 77.1, 0.06; GCF_002884815.1, s__Bifidobacterium sp002884815, 95.0, 77.03, 0.06; GCF_000737845.1, s__Bifidobacterium bombi, 95.0, 77.0, 0.13; GCF_002259795.1, s__Bifidobacterium aquikefiri, 95.0, 76.98, 0.09\", \"msa_percent\": 89.24, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.011.fa\", \"classification\": \"Bacteroidales\", \"fastani_reference\": \"GCF_000012845.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis\", \"fastani_ani\": 97.45, \"fastani_af\": 0.85, \"closest_placement_reference\": \"GCF_000012845.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis\", \"closest_placement_ani\": 97.45, \"closest_placement_af\": 0.85, \"pplacer_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_004793765.1, s__Parabacteroides distasonis_A, 95.0, 94.21, 0.69; GCF_011038785.1, s__Parabacteroides sp011038785, 95.0, 90.1, 0.74; GCF_900155425.1, s__Parabacteroides massiliensis, 95.0, 78.75, 0.23; GCF_000969825.1, s__Parabacteroides gordonii, 95.0, 78.6, 0.22; GCA_900549585.1, s__Parabacteroides sp900549585, 95.0, 78.56, 0.11; GCF_900186615.1, s__Parabacteroides bouchesdurhonensis, 95.0, 78.52, 0.21; GCA_900760525.1, s__Parabacteroides sp900760525, 95.0, 78.48, 0.2; GCF_014287585.1, s__Parabacteroides sp014287585, 95.0, 78.41, 0.23; GCF_900128505.1, s__Parabacteroides timonensis, 95.0, 78.4, 0.23; GCF_900108035.1, s__Parabacteroides chinchillae, 95.0, 78.37, 0.2; GCF_000154105.1, s__Parabacteroides merdae, 95.0, 78.34, 0.22; GCF_003480915.1, s__Parabacteroides sp003480915, 95.0, 78.31, 0.22; GCF_003473295.1, s__Parabacteroides sp003473295, 95.0, 78.31, 0.17; GCF_000156495.1, s__Parabacteroides johnsonii, 95.0, 78.29, 0.21; GCF_000969835.1, s__Parabacteroides goldsteinii, 95.0, 78.28, 0.23; GCA_900540715.1, s__Parabacteroides sp900540715, 95.0, 78.26, 0.22; GCA_014647375.1, s__Parabacteroides faecis, 95.0, 78.25, 0.24; GCF_003363715.1, s__Parabacteroides acidifaciens, 95.0, 78.11, 0.23; GCA_900770835.1, s__Parabacteroides sp900770835, 95.0, 78.07, 0.18; GCA_900541965.1, s__Parabacteroides sp900541965, 95.0, 77.99, 0.17; GCA_000436495.1, s__Parabacteroides sp000436495, 95.0, 77.93, 0.13; GCA_004562445.1, s__Parabacteroides sp004562445, 95.0, 77.73, 0.12; GCA_900552465.1, s__Parabacteroides sp900552465, 95.0, 77.57, 0.19; GCA_900548175.1, s__Parabacteroides sp900548175, 95.0, 77.48, 0.12; GCA_900552415.1, s__Parabacteroides sp900552415, 95.0, 77.33, 0.21; GCF_002159645.1, s__Parabacteroides sp002159645, 95.0, 77.14, 0.1; GCA_900547435.1, s__Parabacteroides sp900547435, 95.0, 76.94, 0.11\", \"msa_percent\": 96.49, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.015.fa\", \"classification\": \"Lachnospirales\", \"fastani_reference\": \"GCF_001406815.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Agathobacter;s__Agathobacter faecis\", \"fastani_ani\": 98.14, \"fastani_af\": 0.77, \"closest_placement_reference\": null, \"closest_placement_radius\": null, \"closest_placement_taxonomy\": null, \"closest_placement_ani\": null, \"closest_placement_af\": null, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Agathobacter;s__\", \"classification_method\": \"ANI\", \"note\": \"topological placement and ANI have incongruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_001406815.1, s__Agathobacter faecis, 95.0, 98.14, 0.77; GCA_900557055.1, s__Agathobacter sp900557055, 95.0, 97.7, 0.52; GCA_002474415.1, s__Agathobacter sp002474415, 95.0, 91.36, 0.7; GCF_902363675.1, s__Agathobacter sp000434275, 95.0, 81.67, 0.39; GCA_900550545.1, s__Agathobacter sp900550545, 95.0, 80.13, 0.33; GCA_900552085.1, s__Agathobacter sp900552085, 95.0, 79.87, 0.34; GCA_900550845.1, s__Agathobacter sp900550845, 95.0, 79.68, 0.13; GCF_000020605.1, s__Agathobacter rectalis, 95.0, 79.33, 0.16; GCA_900543445.1, s__Agathobacter sp900543445, 95.0, 79.24, 0.32; GCA_900549895.1, s__Agathobacter sp900549895, 95.0, 79.2, 0.36; GCA_900547695.1, s__Agathobacter sp900547695, 95.0, 78.91, 0.15; GCA_900546625.1, s__Agathobacter sp900546625, 95.0, 78.64, 0.16; GCA_900316805.1, s__Agathobacter sp900316805, 95.0, 78.16, 0.28; GCA_900317585.1, s__Agathobacter sp900317585, 95.0, 78.1, 0.15; GCA_900548765.1, s__Agathobacter sp900548765, 95.0, 77.9, 0.13; GCF_002735305.1, s__Agathobacter ruminis, 95.0, 77.1, 0.11\", \"msa_percent\": 63.25, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.016.fa\", \"classification\": \"Acidaminococcales\", \"fastani_reference\": \"GCF_900095825.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus massiliensis\", \"fastani_ani\": 98.6, \"fastani_af\": 0.89, \"closest_placement_reference\": \"GCF_900095825.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus massiliensis\", \"closest_placement_ani\": 98.6, \"closest_placement_af\": 0.89, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_004555735.1, s__Acidaminococcus fermentans_A, 95.0, 88.82, 0.85; GCF_009696675.1, s__Acidaminococcus sp000437815, 95.0, 87.34, 0.69; GCF_000025305.1, s__Acidaminococcus fermentans, 95.0, 86.56, 0.71; GCF_900106585.1, s__Acidaminococcus timonensis, 95.0, 83.24, 0.63; GCA_900770265.1, s__Acidaminococcus sp900770265, 95.0, 83.0, 0.71; GCF_900291475.1, s__Acidaminococcus provencensis, 95.0, 80.54, 0.43; GCA_900314165.1, s__Acidaminococcus sp900314165, 95.0, 79.11, 0.34; GCA_900315205.1, s__Acidaminococcus sp900315205, 95.0, 78.49, 0.28; GCA_900554515.1, s__Acidaminococcus sp900554515, 95.0, 78.14, 0.28; GCA_900538365.1, s__Acidaminococcus sp900538365, 95.0, 77.58, 0.17; GCA_902796945.1, s__Acidaminococcus sp902796945, 95.0, 77.55, 0.23; GCF_000425045.1, s__Acidaminococcus intestini, 95.0, 77.39, 0.16\", \"msa_percent\": 78.94, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.018.fa\", \"classification\": \"Oscillospirales\", \"fastani_reference\": null, \"fastani_reference_radius\": null, \"fastani_taxonomy\": null, \"fastani_ani\": null, \"fastani_af\": null, \"closest_placement_reference\": \"GCA_900540775.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Gemmiger;s__Gemmiger sp900540775\", \"closest_placement_ani\": 86.46, \"closest_placement_af\": 0.73, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Gemmiger;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": null, \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_900167555.1, s__Gemmiger formicilis, 95.0, 87.25, 0.77; GCA_900540595.1, s__Gemmiger sp900540595, 95.0, 86.2, 0.75; GCA_900554145.1, s__Gemmiger sp900554145, 95.0, 84.63, 0.76; GCA_004554775.1, s__Gemmiger variabilis_B, 95.0, 82.86, 0.68; GCA_900539695.1, s__Gemmiger sp900539695, 95.0, 82.63, 0.61; GCA_900772715.1, s__Gemmiger sp900772715, 95.0, 82.4, 0.59; GCA_004552305.1, s__Gemmiger variabilis_A, 95.0, 81.71, 0.64; GCA_900545545.1, s__Gemmiger sp900545545, 95.0, 80.79, 0.49; GCA_003343905.1, s__Gemmiger variabilis_C, 95.0, 80.6, 0.5; GCF_000157955.1, s__Gemmiger variabilis, 95.0, 80.55, 0.48; GCA_900556255.1, s__Gemmiger sp900556255, 95.0, 80.51, 0.65; GCF_003324125.1, s__Gemmiger qucibialis, 95.0, 80.26, 0.43; GCA_004561545.1, s__Gemmiger sp004561545, 95.0, 79.62, 0.42; GCA_002306375.1, s__Gemmiger sp002306375, 95.0, 79.36, 0.43; GCA_900548355.1, s__Gemmiger sp900548355, 95.0, 78.68, 0.33; GCA_004555405.1, s__Gemmiger sp004555405, 95.0, 77.64, 0.17\", \"msa_percent\": 92.75, \"translation_table\": 11.0, \"red_value\": 0.996763018207208, \"warnings\": \"Genome not assigned to closest species as it falls outside its pre-defined ANI radius\"}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.019.fa\", \"classification\": \"Lachnospirales\", \"fastani_reference\": \"GCF_002234575.2\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Enterocloster;s__Enterocloster bolteae\", \"fastani_ani\": 98.97, \"fastani_af\": 0.93, \"closest_placement_reference\": \"GCF_002234575.2\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Enterocloster;s__Enterocloster bolteae\", \"closest_placement_ani\": 98.97, \"closest_placement_af\": 0.93, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Enterocloster;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000424325.1, s__Enterocloster clostridioformis_A, 95.0, 92.13, 0.63; GCF_900113155.1, s__Enterocloster clostridioformis, 95.0, 91.44, 0.72; GCF_005845215.1, s__Enterocloster sp005845215, 95.0, 88.53, 0.85; GCF_003434055.1, s__Enterocloster aldenensis, 95.0, 80.55, 0.46; GCF_000155435.1, s__Enterocloster sp000155435, 95.0, 80.46, 0.47; GCF_000233455.1, s__Enterocloster citroniae, 95.0, 80.0, 0.43; GCF_003473545.1, s__Enterocloster sp000431375, 95.0, 79.15, 0.11; GCF_900753815.1, s__Enterocloster sp900753815, 95.0, 78.73, 0.25; GCF_900770345.1, s__Enterocloster sp900770345, 95.0, 78.68, 0.17; GCA_001304855.1, s__Enterocloster sp001304855, 95.0, 78.65, 0.31; GCA_900555045.1, s__Enterocloster sp900555045, 95.0, 78.6, 0.21; GCF_013282095.1, s__Enterocloster sp900538485, 95.0, 78.38, 0.16; GCF_001517625.2, s__Enterocloster sp001517625, 95.0, 78.33, 0.15; GCF_009696375.1, s__Enterocloster sp900540675, 95.0, 78.29, 0.19; GCA_900549235.1, s__Enterocloster sp900549235, 95.0, 78.27, 0.21; GCA_900551225.1, s__Enterocloster sp900551225, 95.0, 78.06, 0.23; GCF_902364025.1, s__Enterocloster lavalensis, 95.0, 77.86, 0.28; GCF_000158075.1, s__Enterocloster asparagiformis, 95.0, 77.81, 0.25; GCA_900547035.1, s__Enterocloster sp900547035, 95.0, 77.8, 0.14; GCA_900543885.1, s__Enterocloster sp900543885, 95.0, 77.76, 0.12; GCF_900759225.1, s__Enterocloster sp900759225, 95.0, 77.66, 0.22; GCA_900555905.1, s__Enterocloster sp900555905, 95.0, 77.58, 0.19; GCA_900541315.1, s__Enterocloster sp900541315, 95.0, 77.56, 0.12\", \"msa_percent\": 66.09, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": \"Genome has more than 11.7% of markers with multiple hits\"}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.021.fa\", \"classification\": \"Acidaminococcales\", \"fastani_reference\": \"GCF_900291475.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus provencensis\", \"fastani_ani\": 99.52, \"fastani_af\": 0.96, \"closest_placement_reference\": \"GCF_900291475.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus provencensis\", \"closest_placement_ani\": 99.52, \"closest_placement_af\": 0.96, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_009696675.1, s__Acidaminococcus sp000437815, 95.0, 80.18, 0.5; GCF_000025305.1, s__Acidaminococcus fermentans, 95.0, 80.03, 0.54; GCA_900770265.1, s__Acidaminococcus sp900770265, 95.0, 80.02, 0.5; GCF_900095825.1, s__Acidaminococcus massiliensis, 95.0, 79.75, 0.48; GCF_900106585.1, s__Acidaminococcus timonensis, 95.0, 79.74, 0.5; GCA_004555735.1, s__Acidaminococcus fermentans_A, 95.0, 79.28, 0.43; GCA_900314165.1, s__Acidaminococcus sp900314165, 95.0, 79.06, 0.34; GCA_902796945.1, s__Acidaminococcus sp902796945, 95.0, 77.86, 0.21; GCA_900315205.1, s__Acidaminococcus sp900315205, 95.0, 77.59, 0.2; GCA_900538365.1, s__Acidaminococcus sp900538365, 95.0, 77.53, 0.18; GCF_000425045.1, s__Acidaminococcus intestini, 95.0, 77.39, 0.16; GCA_900554515.1, s__Acidaminococcus sp900554515, 95.0, 77.37, 0.27\", \"msa_percent\": 84.0, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.022.fa\", \"classification\": \"Coriobacteriales\", \"fastani_reference\": null, \"fastani_reference_radius\": null, \"fastani_taxonomy\": null, \"fastani_ani\": null, \"fastani_af\": null, \"closest_placement_reference\": \"GCF_009695875.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Coriobacteriia;o__Coriobacteriales;f__Atopobiaceae;g__Olsenella;s__Olsenella sp900555995\", \"closest_placement_ani\": 92.56, \"closest_placement_af\": 0.89, \"pplacer_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Coriobacteriia;o__Coriobacteriales;f__Atopobiaceae;g__Olsenella;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": null, \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_003862195.1, s__Olsenella sp003862195, 95.0, 81.21, 0.5; GCA_902787335.1, s__Olsenella sp902787335, 95.0, 81.13, 0.53; GCA_900313175.1, s__Olsenella sp900313175, 95.0, 81.13, 0.53; GCF_012844235.1, s__Olsenella sp900538935, 95.0, 80.97, 0.51; GCF_900120385.1, s__Olsenella phocaeensis, 95.0, 80.9, 0.45; GCF_900105025.1, s__Olsenella umbonata, 95.0, 80.71, 0.47; GCF_000143845.1, s__Olsenella uli, 95.0, 80.13, 0.44; GCF_900155635.1, s__Olsenella urininfantis, 95.0, 80.09, 0.41; GCF_001457795.1, s__Olsenella sp001457795, 95.0, 79.91, 0.43; GCF_000468755.1, s__Olsenella profusa, 95.0, 79.87, 0.38; GCF_900111255.1, s__Olsenella umbonata_A, 95.0, 79.86, 0.39; GCA_900316105.1, s__Olsenella sp900316105, 95.0, 79.44, 0.32; GCF_001189515.2, s__Olsenella sp001189515, 95.0, 78.06, 0.27\", \"msa_percent\": 95.37, \"translation_table\": 11.0, \"red_value\": 0.9944014303711676, \"warnings\": \"Genome not assigned to closest species as it falls outside its pre-defined ANI radius\"}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.024.fa\", \"classification\": \"Burkholderiales\", \"fastani_reference\": \"GCF_003315195.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae;g__Sutterella;s__Sutterella wadsworthensis\", \"fastani_ani\": 97.78, \"fastani_af\": 0.86, \"closest_placement_reference\": \"GCF_003315195.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae;g__Sutterella;s__Sutterella wadsworthensis\", \"closest_placement_ani\": 97.78, \"closest_placement_af\": 0.86, \"pplacer_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae;g__Sutterella;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_009183815.1, s__Sutterella seckii_A, 95.0, 79.32, 0.31; GCF_006337085.1, s__Sutterella faecalis, 95.0, 79.09, 0.31; GCA_900770335.1, s__Sutterella sp900770335, 95.0, 78.61, 0.27; GCF_000250875.1, s__Sutterella parvirubra, 95.0, 78.52, 0.17; GCF_003609995.1, s__Sutterella megalosphaeroides, 95.0, 78.31, 0.17; GCA_900754475.1, s__Sutterella sp900754475, 95.0, 78.11, 0.19; GCF_000297775.1, s__Sutterella wadsworthensis_A, 95.0, 78.08, 0.15; GCA_900543805.1, s__Sutterella sp900543805, 95.0, 77.95, 0.17; GCA_900545275.1, s__Sutterella sp900545275, 95.0, 77.92, 0.15; GCA_900757865.1, s__Sutterella sp900757865, 95.0, 77.91, 0.06; GCA_900764215.1, s__Sutterella sp900764215, 95.0, 77.83, 0.17; GCA_900762445.1, s__Sutterella sp900762445, 95.0, 77.07, 0.08\", \"msa_percent\": 83.44, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": \"Genome has more than 10.0% of markers with multiple hits\"}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.025.fa\", \"classification\": \"Oscillospirales\", \"fastani_reference\": \"GCA_000436335.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__CAG-217;s__CAG-217 sp000436335\", \"fastani_ani\": 97.27, \"fastani_af\": 0.85, \"closest_placement_reference\": \"GCA_000436335.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__CAG-217;s__CAG-217 sp000436335\", \"closest_placement_ani\": 97.27, \"closest_placement_af\": 0.85, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__CAG-217;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_900547275.1, s__CAG-217 sp900547275, 95.0, 77.29, 0.13\", \"msa_percent\": 83.68, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.027.fa\", \"classification\": \"Oscillospirales\", \"fastani_reference\": \"GCF_010509575.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii_C\", \"fastani_ani\": 97.44, \"fastani_af\": 0.89, \"closest_placement_reference\": \"GCF_010509575.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii_C\", \"closest_placement_ani\": 97.44, \"closest_placement_af\": 0.89, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_002550035.1, s__Faecalibacterium prausnitzii_E, 95.0, 86.23, 0.81; GCF_002549975.1, s__Faecalibacterium prausnitzii_H, 95.0, 85.82, 0.83; GCF_003287495.1, s__Faecalibacterium prausnitzii_I, 95.0, 85.18, 0.76; GCA_900539945.1, s__Faecalibacterium sp900539945, 95.0, 85.16, 0.75; GCF_002549775.1, s__Faecalibacterium prausnitzii_F, 95.0, 85.05, 0.78; GCA_900539885.1, s__Faecalibacterium sp900539885, 95.0, 84.93, 0.75; GCF_002549755.1, s__Faecalibacterium prausnitzii_D, 95.0, 84.81, 0.77; GCF_003324185.1, s__Faecalibacterium prausnitzii, 95.0612, 84.77, 0.77; GCF_003287405.1, s__Faecalibacterium prausnitzii_J, 95.0, 84.67, 0.76; GCF_003293635.1, s__Faecalibacterium prausnitzii_G, 95.0, 84.59, 0.76; GCF_002550015.1, s__Faecalibacterium prausnitzii_A, 95.0612, 84.46, 0.78; GCA_900765705.1, s__Faecalibacterium sp900765705, 95.0, 84.43, 0.57; GCA_900772565.1, s__Faecalibacterium sp900772565, 95.0, 84.35, 0.7; GCA_900758465.1, s__Faecalibacterium sp900758465, 95.0, 84.22, 0.72; GCA_003449675.1, s__Faecalibacterium sp003449675, 95.0, 84.09, 0.56; GCA_900551435.1, s__Faecalibacterium sp900551435, 95.0, 84.0, 0.75; GCA_900765105.1, s__Faecalibacterium sp900765105, 95.0, 83.42, 0.64; GCA_004558805.1, s__Faecalibacterium prausnitzii_M, 95.0, 83.22, 0.74; GCA_002313795.1, s__Faecalibacterium prausnitzii_L, 95.0, 83.18, 0.61; GCA_900540455.1, s__Faecalibacterium sp900540455, 95.0, 81.81, 0.6; GCF_002160895.1, s__Faecalibacterium sp002160895, 95.0, 81.45, 0.58; GCF_002160915.1, s__Faecalibacterium sp002160915, 95.0, 80.86, 0.58\", \"msa_percent\": 76.57, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.028.fa\", \"classification\": \"Bacteroidales\", \"fastani_reference\": null, \"fastani_reference_radius\": null, \"fastani_taxonomy\": null, \"fastani_ani\": null, \"fastani_af\": null, \"closest_placement_reference\": \"GCA_900765785.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides sp900765785\", \"closest_placement_ani\": 87.55, \"closest_placement_af\": 0.14, \"pplacer_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": null, \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_900557355.1, s__Bacteroides sp900557355, 95.0, 92.43, 0.19; GCF_002222615.2, s__Bacteroides caccae, 95.0, 90.72, 0.17; GCF_000011065.1, s__Bacteroides thetaiotaomicron, 95.0, 90.39, 0.26; GCF_001314995.1, s__Bacteroides ovatus, 95.0, 89.42, 0.23; GCA_900761785.1, s__Bacteroides sp900761785, 95.0, 88.76, 0.3; GCF_014334015.1, s__Bacteroides sp003463205, 95.0, 87.7, 0.25; GCA_900556625.1, s__Bacteroides sp900556625, 95.0, 87.04, 0.12; GCF_009193295.2, s__Bacteroides sp007097645, 95.0, 85.47, 0.18; GCA_000210075.1, s__Bacteroides xylanisolvens, 95.0, 85.43, 0.18; GCF_902362375.1, s__Bacteroides sp902362375, 95.0, 84.91, 0.22; GCA_900755095.1, s__Bacteroides sp900755095, 95.0, 84.44, 0.1; GCA_900555635.1, s__Bacteroides sp900555635, 95.0, 83.35, 0.05; GCF_000156195.1, s__Bacteroides finegoldii, 95.8579, 83.3, 0.14; GCF_001688725.2, s__Bacteroides caecimuris, 95.0, 82.78, 0.11; GCF_900130125.1, s__Bacteroides congonensis, 95.0, 82.55, 0.17; GCF_000613385.1, s__Bacteroides acidifaciens, 95.0, 82.31, 0.09; GCF_000374365.1, s__Bacteroides gallinarum, 95.0, 82.26, 0.02; GCF_012113595.1, s__Bacteroides sp012113595, 95.0, 82.23, 0.2; GCF_900130135.1, s__Bacteroides togonis, 95.0, 82.09, 0.02; GCF_903181435.1, s__Bacteroides acidifaciens_A, 95.0, 81.87, 0.13; GCF_900106755.1, s__Bacteroides faecis, 95.0, 81.79, 0.19; GCF_009193325.2, s__Bacteroides sp900066265, 95.8579, 81.79, 0.16; GCF_900108345.1, s__Bacteroides ndongoniae, 95.0, 81.46, 0.02; GCF_000172175.1, s__Bacteroides intestinalis, 95.0, 81.36, 0.06; GCF_002998435.1, s__Bacteroides zoogleoformans, 95.0, 81.27, 0.03; GCF_000613465.1, s__Bacteroides nordii, 95.0, 80.66, 0.04; GCF_902364365.1, s__Bacteroides sp900556215, 95.2143, 80.58, 0.07; GCF_000158035.1, s__Bacteroides cellulosilyticus, 95.0, 80.25, 0.04; GCA_900760255.1, s__Bacteroides sp900760255, 95.0, 80.12, 0.05; GCF_900128475.1, s__Bacteroides massiliensis_A, 95.0, 80.09, 0.02; GCF_003464595.1, s__Bacteroides intestinalis_A, 95.2143, 79.97, 0.06; GCF_000315485.1, s__Bacteroides oleiciplenus, 95.0, 79.81, 0.05; GCF_900241005.1, s__Bacteroides cutis, 95.0, 79.6, 0.04; GCF_002849695.1, s__Bacteroides fragilis_A, 95.0, 79.6, 0.05; GCA_900552405.1, s__Bacteroides sp900552405, 95.0, 79.59, 0.05; GCA_900547205.1, s__Bacteroides sp900547205, 95.0, 79.55, 0.09; GCF_014196225.1, s__Bacteroides pyogenes_A, 95.0, 79.45, 0.04; GCF_000195635.1, s__Bacteroides fluxus, 95.0, 79.24, 0.04; GCF_000155815.1, s__Bacteroides eggerthii, 95.0, 79.11, 0.04; GCF_000513195.1, s__Bacteroides timonensis, 95.0, 79.1, 0.04; GCF_003865075.1, s__Bacteroides sp003865075, 95.0, 78.88, 0.08; GCA_902388495.1, s__Bacteroides sp902388495, 95.0, 78.74, 0.04; GCF_000614125.1, s__Bacteroides rodentium, 95.0, 78.68, 0.05; GCF_900155865.1, s__Bacteroides bouchesdurhonensis, 95.0, 78.64, 0.07; GCA_900553815.1, s__Bacteroides sp900553815, 95.0, 78.45, 0.01; GCF_000499785.1, s__Bacteroides neonati, 95.0, 78.41, 0.04; GCF_000025985.1, s__Bacteroides fragilis, 95.0, 78.33, 0.05; GCF_000154525.1, s__Bacteroides stercoris, 95.0, 78.31, 0.04; GCF_003438615.1, s__Bacteroides sp003545565, 95.0, 78.23, 0.04; GCF_000428105.1, s__Bacteroides pyogenes, 95.0, 78.21, 0.02; GCF_000381365.1, s__Bacteroides salyersiae, 95.0, 78.05, 0.06; GCA_900766005.1, s__Bacteroides sp900766005, 95.0, 77.89, 0.03; GCF_004793475.1, s__Bacteroides sp002491635, 95.0, 77.87, 0.05; GCF_000614145.1, s__Bacteroides faecichinchillae, 95.0, 77.81, 0.08; GCA_900765805.1, s__Bacteroides sp900765805, 95.0, 77.76, 0.04; GCA_014385165.1, s__Bacteroides sp014385165, 95.0, 77.62, 0.03; GCA_900762525.1, s__Bacteroides sp900762525, 95.0, 77.48, 0.06; GCF_000186225.1, s__Bacteroides helcogenes, 95.0, 77.43, 0.03; GCF_000614165.1, s__Bacteroides stercorirosoris, 95.0, 77.28, 0.05; GCA_002293435.1, s__Bacteroides sp002293435, 95.0, 77.22, 0.02; GCF_004342845.1, s__Bacteroides heparinolyticus, 95.0, 77.14, 0.02; GCF_000517545.1, s__Bacteroides reticulotermitis, 95.0, 77.05, 0.04; GCF_900129655.1, s__Bacteroides clarus, 95.0, 77.01, 0.03; GCF_000154205.1, s__Bacteroides uniformis, 95.0, 76.86, 0.04; GCA_009929715.1, s__Bacteroides sp009929715, 95.0, 76.73, 0.01; GCF_900104585.1, s__Bacteroides ihuae, 95.0, 76.68, 0.01; GCA_002307035.1, s__Bacteroides sp002307035, 95.0, 76.49, 0.02; GCF_010500965.1, s__Bacteroides sp010500965, 95.0, 76.35, 0.02; GCF_007896885.1, s__Bacteroides sp007896885, 95.0, 75.98, 0.01; GCF_000428125.1, s__Bacteroides graminisolvens, 95.0, 75.72, 0.0; GCF_010500955.1, s__Bacteroides sp010500955, 95.0, 75.54, 0.03; GCF_002160055.1, s__Bacteroides sp002160055, 95.0, 75.43, 0.01; GCF_010500995.1, s__Bacteroides sp010500995, 95.0, 75.39, 0.03; GCA_900766195.1, s__Bacteroides sp900766195, 95.0, 75.27, 0.01; GCA_002471185.1, s__Bacteroides sp002471185, 95.0, 75.03, 0.01; GCA_002471195.1, s__Bacteroides sp002471195, 95.0, 74.6, 0.01; GCF_010501015.1, s__Bacteroides sp010501015, 95.0, 74.6, 0.0\", \"msa_percent\": 27.58, \"translation_table\": 11.0, \"red_value\": 0.9987817030090164, \"warnings\": \"Genome has more than 11.7% of markers with multiple hits\"}, {\"user_genome\": \"MEGAHIT-MaxBin2-CAPES_S11.031.fa\", \"classification\": \"Veillonellales\", \"fastani_reference\": \"GCF_003467125.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Caecibacter;s__Caecibacter sp003467125\", \"fastani_ani\": 98.19, \"fastani_af\": 0.69, \"closest_placement_reference\": \"GCF_003467125.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Caecibacter;s__Caecibacter sp003467125\", \"closest_placement_ani\": 98.19, \"closest_placement_af\": 0.69, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Caecibacter;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_900095875.1, s__Caecibacter massiliensis, 95.0, 87.68, 0.68; GCA_900752195.1, s__Caecibacter sp900549385, 95.0, 78.72, 0.15; GCF_014269395.1, s__Caecibacter hominis, 95.0, 78.36, 0.29\", \"msa_percent\": 33.93, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": \"Genome has more than 19.2% of markers with multiple hits\"}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.11.fa\", \"classification\": \"Lachnospirales\", \"fastani_reference\": \"GCF_009680455.1\", \"fastani_reference_radius\": 95.6336, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Lachnospira;s__Lachnospira eligens_A\", \"fastani_ani\": 97.74, \"fastani_af\": 0.97, \"closest_placement_reference\": \"GCF_009680455.1\", \"closest_placement_radius\": 95.6336, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Lachnospira;s__Lachnospira eligens_A\", \"closest_placement_ani\": 97.74, \"closest_placement_af\": 0.97, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Lachnospira;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000146185.1, s__Lachnospira eligens, 95.6336, 95.82, 0.94; GCF_003464165.1, s__Lachnospira sp003451515, 95.0, 95.14, 0.92; GCF_003458705.1, s__Lachnospira sp003537285, 95.4867, 94.47, 0.94; GCA_900547255.1, s__Lachnospira sp900547255, 95.0, 79.51, 0.43; GCA_900112995.1, s__Lachnospira rogosae_A, 95.0, 79.46, 0.45; GCF_000702205.1, s__Lachnospira multipara_A, 95.0, 79.42, 0.19; GCA_900772425.1, s__Lachnospira sp900772425, 95.0, 79.3, 0.26; GCF_000424105.1, s__Lachnospira multipara, 95.5522, 79.18, 0.21; GCA_000436475.1, s__Lachnospira sp000436475, 95.0, 79.06, 0.43; GCA_000436535.1, s__Lachnospira sp000436535, 95.0, 78.98, 0.37; GCF_014287955.1, s__Lachnospira sp900316325, 95.0, 78.98, 0.38; GCA_002435585.1, s__Lachnospira sp002435585, 95.0, 78.92, 0.34; GCF_900103815.1, s__Lachnospira pectinoschiza_B, 95.5522, 78.84, 0.22; GCA_900552795.1, s__Lachnospira sp900552795, 95.0, 78.07, 0.17; GCA_002394205.1, s__Lachnospira sp002394205, 95.0, 77.94, 0.18; GCA_900545725.1, s__Lachnospira sp900545725, 95.0, 77.92, 0.34; GCA_000437735.1, s__Lachnospira sp000437735, 95.0, 77.46, 0.22; GCA_900551945.1, s__Lachnospira sp900551945, 95.0, 77.09, 0.24\", \"msa_percent\": 54.32, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.13.fa\", \"classification\": \"Lachnospirales\", \"fastani_reference\": \"GCF_001406815.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Agathobacter;s__Agathobacter faecis\", \"fastani_ani\": 98.19, \"fastani_af\": 0.9, \"closest_placement_reference\": null, \"closest_placement_radius\": null, \"closest_placement_taxonomy\": null, \"closest_placement_ani\": null, \"closest_placement_af\": null, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Agathobacter;s__\", \"classification_method\": \"ANI\", \"note\": \"topological placement and ANI have incongruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_001406815.1, s__Agathobacter faecis, 95.0, 98.19, 0.9; GCA_900557055.1, s__Agathobacter sp900557055, 95.0, 97.79, 0.57; GCA_002474415.1, s__Agathobacter sp002474415, 95.0, 91.41, 0.66; GCF_902363675.1, s__Agathobacter sp000434275, 95.0, 81.59, 0.44; GCA_900550845.1, s__Agathobacter sp900550845, 95.0, 80.66, 0.16; GCA_900550545.1, s__Agathobacter sp900550545, 95.0, 80.01, 0.39; GCA_900552085.1, s__Agathobacter sp900552085, 95.0, 79.9, 0.33; GCA_900543445.1, s__Agathobacter sp900543445, 95.0, 79.14, 0.34; GCA_900549895.1, s__Agathobacter sp900549895, 95.0, 79.04, 0.33; GCA_900316805.1, s__Agathobacter sp900316805, 95.0, 78.26, 0.25; GCF_000020605.1, s__Agathobacter rectalis, 95.0, 78.15, 0.17; GCA_900546625.1, s__Agathobacter sp900546625, 95.0, 77.95, 0.17; GCA_900547695.1, s__Agathobacter sp900547695, 95.0, 77.94, 0.16; GCA_900548765.1, s__Agathobacter sp900548765, 95.0, 77.83, 0.11; GCA_900317585.1, s__Agathobacter sp900317585, 95.0, 77.41, 0.14; GCF_002735305.1, s__Agathobacter ruminis, 95.0, 77.02, 0.13\", \"msa_percent\": 61.56, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.15.fa\", \"classification\": \"Bacteroidales\", \"fastani_reference\": \"GCF_000012845.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis\", \"fastani_ani\": 97.67, \"fastani_af\": 0.91, \"closest_placement_reference\": \"GCF_000012845.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__Parabacteroides distasonis\", \"closest_placement_ani\": 97.67, \"closest_placement_af\": 0.91, \"pplacer_taxonomy\": \"d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Tannerellaceae;g__Parabacteroides;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_004793765.1, s__Parabacteroides distasonis_A, 95.0, 94.58, 0.7; GCF_011038785.1, s__Parabacteroides sp011038785, 95.0, 90.43, 0.6; GCA_900549585.1, s__Parabacteroides sp900549585, 95.0, 79.5, 0.09; GCF_000969825.1, s__Parabacteroides gordonii, 95.0, 78.95, 0.27; GCF_900108035.1, s__Parabacteroides chinchillae, 95.0, 78.78, 0.2; GCF_900186615.1, s__Parabacteroides bouchesdurhonensis, 95.0, 78.76, 0.21; GCF_900155425.1, s__Parabacteroides massiliensis, 95.0, 78.73, 0.24; GCA_014647375.1, s__Parabacteroides faecis, 95.0, 78.66, 0.3; GCF_000154105.1, s__Parabacteroides merdae, 95.0, 78.63, 0.24; GCF_000969835.1, s__Parabacteroides goldsteinii, 95.0, 78.58, 0.27; GCF_003473295.1, s__Parabacteroides sp003473295, 95.0, 78.54, 0.21; GCF_000156495.1, s__Parabacteroides johnsonii, 95.0, 78.49, 0.25; GCA_900760525.1, s__Parabacteroides sp900760525, 95.0, 78.42, 0.24; GCF_014287585.1, s__Parabacteroides sp014287585, 95.0, 78.41, 0.28; GCF_900128505.1, s__Parabacteroides timonensis, 95.0, 78.39, 0.26; GCF_003480915.1, s__Parabacteroides sp003480915, 95.0, 78.38, 0.28; GCF_003363715.1, s__Parabacteroides acidifaciens, 95.0, 78.2, 0.29; GCA_900770835.1, s__Parabacteroides sp900770835, 95.0, 78.15, 0.17; GCA_900540715.1, s__Parabacteroides sp900540715, 95.0, 78.11, 0.26; GCA_900541965.1, s__Parabacteroides sp900541965, 95.0, 78.0, 0.22; GCA_000436495.1, s__Parabacteroides sp000436495, 95.0, 77.94, 0.13; GCA_900552465.1, s__Parabacteroides sp900552465, 95.0, 77.54, 0.23; GCA_004562445.1, s__Parabacteroides sp004562445, 95.0, 77.44, 0.08; GCF_002159645.1, s__Parabacteroides sp002159645, 95.0, 77.43, 0.09; GCA_900548175.1, s__Parabacteroides sp900548175, 95.0, 77.38, 0.12; GCA_900552415.1, s__Parabacteroides sp900552415, 95.0, 77.32, 0.14; GCA_900547435.1, s__Parabacteroides sp900547435, 95.0, 76.78, 0.11\", \"msa_percent\": 42.92, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.16.fa\", \"classification\": \"Acidaminococcales\", \"fastani_reference\": \"GCF_900095825.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus massiliensis\", \"fastani_ani\": 98.68, \"fastani_af\": 0.9, \"closest_placement_reference\": \"GCF_900095825.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus massiliensis\", \"closest_placement_ani\": 98.68, \"closest_placement_af\": 0.9, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_004555735.1, s__Acidaminococcus fermentans_A, 95.0, 88.76, 0.84; GCF_009696675.1, s__Acidaminococcus sp000437815, 95.0, 87.43, 0.71; GCF_000025305.1, s__Acidaminococcus fermentans, 95.0, 86.55, 0.72; GCF_900106585.1, s__Acidaminococcus timonensis, 95.0, 83.21, 0.64; GCA_900770265.1, s__Acidaminococcus sp900770265, 95.0, 82.98, 0.7; GCF_900291475.1, s__Acidaminococcus provencensis, 95.0, 80.43, 0.43; GCA_900314165.1, s__Acidaminococcus sp900314165, 95.0, 79.1, 0.35; GCA_900315205.1, s__Acidaminococcus sp900315205, 95.0, 78.48, 0.28; GCA_900554515.1, s__Acidaminococcus sp900554515, 95.0, 78.14, 0.28; GCA_902796945.1, s__Acidaminococcus sp902796945, 95.0, 77.55, 0.23; GCA_900538365.1, s__Acidaminococcus sp900538365, 95.0, 77.46, 0.17; GCF_000425045.1, s__Acidaminococcus intestini, 95.0, 77.27, 0.16\", \"msa_percent\": 78.94, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.22.fa\", \"classification\": \"Oscillospirales\", \"fastani_reference\": \"GCA_000436335.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__CAG-217;s__CAG-217 sp000436335\", \"fastani_ani\": 97.26, \"fastani_af\": 0.91, \"closest_placement_reference\": \"GCA_000436335.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__CAG-217;s__CAG-217 sp000436335\", \"closest_placement_ani\": 97.26, \"closest_placement_af\": 0.91, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Acutalibacteraceae;g__CAG-217;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_900547275.1, s__CAG-217 sp900547275, 95.0, 77.25, 0.14\", \"msa_percent\": 87.0, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.24.fa\", \"classification\": \"Veillonellales\", \"fastani_reference\": \"GCF_000417505.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Megasphaera;s__Megasphaera sp000417505\", \"fastani_ani\": 99.37, \"fastani_af\": 0.92, \"closest_placement_reference\": \"GCF_000417505.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Megasphaera;s__Megasphaera sp000417505\", \"closest_placement_ani\": 99.37, \"closest_placement_af\": 0.92, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Veillonellales;f__Megasphaeraceae;g__Megasphaera;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_902388315.1, s__Megasphaera sp902388315, 95.0, 91.92, 0.76; GCF_003010495.1, s__Megasphaera elsdenii, 95.0, 90.94, 0.89; GCA_900548635.1, s__Megasphaera sp900548635, 95.0, 87.28, 0.86; GCF_001546855.1, s__Megasphaera sp001546855, 95.0, 82.97, 0.72; GCA_900539295.1, s__Megasphaera sp900539295, 95.0, 82.95, 0.7; GCF_000455225.1, s__Megasphaera massiliensis, 95.0, 82.54, 0.71; GCA_002319965.1, s__Megasphaera sp002319965, 95.0, 82.27, 0.62; GCF_003367905.1, s__Megasphaera stantonii, 95.0, 79.75, 0.51; GCA_900554215.1, s__Megasphaera sp900554215, 95.0, 79.22, 0.43; GCA_002431345.1, s__Megasphaera sp002431345, 95.0, 78.27, 0.27; GCA_900540735.1, s__Megasphaera sp900540735, 95.0, 77.99, 0.31; GCF_001045675.1, s__Megasphaera cerevisiae, 95.0, 77.81, 0.18\", \"msa_percent\": 81.93, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.25.fa\", \"classification\": \"Lachnospirales\", \"fastani_reference\": \"GCF_002234575.2\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Enterocloster;s__Enterocloster bolteae\", \"fastani_ani\": 99.06, \"fastani_af\": 0.98, \"closest_placement_reference\": \"GCF_002234575.2\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Enterocloster;s__Enterocloster bolteae\", \"closest_placement_ani\": 99.06, \"closest_placement_af\": 0.98, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__Enterocloster;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000424325.1, s__Enterocloster clostridioformis_A, 95.0, 92.17, 0.68; GCF_900113155.1, s__Enterocloster clostridioformis, 95.0, 91.6, 0.76; GCF_005845215.1, s__Enterocloster sp005845215, 95.0, 88.6, 0.89; GCF_000155435.1, s__Enterocloster sp000155435, 95.0, 80.39, 0.49; GCF_003434055.1, s__Enterocloster aldenensis, 95.0, 80.27, 0.47; GCF_000233455.1, s__Enterocloster citroniae, 95.0, 79.87, 0.45; GCF_900753815.1, s__Enterocloster sp900753815, 95.0, 78.86, 0.23; GCF_900770345.1, s__Enterocloster sp900770345, 95.0, 78.7, 0.19; GCA_900555045.1, s__Enterocloster sp900555045, 95.0, 78.65, 0.22; GCA_001304855.1, s__Enterocloster sp001304855, 95.0, 78.56, 0.32; GCF_003473545.1, s__Enterocloster sp000431375, 95.0, 78.46, 0.1; GCF_009696375.1, s__Enterocloster sp900540675, 95.0, 78.41, 0.19; GCA_900549235.1, s__Enterocloster sp900549235, 95.0, 78.32, 0.2; GCF_001517625.2, s__Enterocloster sp001517625, 95.0, 78.08, 0.15; GCF_013282095.1, s__Enterocloster sp900538485, 95.0, 78.06, 0.15; GCA_900551225.1, s__Enterocloster sp900551225, 95.0, 77.99, 0.24; GCF_000158075.1, s__Enterocloster asparagiformis, 95.0, 77.81, 0.27; GCF_902364025.1, s__Enterocloster lavalensis, 95.0, 77.8, 0.3; GCA_900547035.1, s__Enterocloster sp900547035, 95.0, 77.77, 0.13; GCA_900543885.1, s__Enterocloster sp900543885, 95.0, 77.66, 0.11; GCA_900555905.1, s__Enterocloster sp900555905, 95.0, 77.56, 0.2; GCA_900541315.1, s__Enterocloster sp900541315, 95.0, 77.55, 0.1; GCF_900759225.1, s__Enterocloster sp900759225, 95.0, 77.55, 0.2\", \"msa_percent\": 63.01, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.26.fa\", \"classification\": \"Enterobacterales\", \"fastani_reference\": \"GCF_002950215.1\", \"fastani_reference_radius\": 97.0028, \"fastani_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia;s__Escherichia flexneri\", \"fastani_ani\": 97.95, \"fastani_af\": 0.89, \"closest_placement_reference\": \"GCF_002950215.1\", \"closest_placement_radius\": 97.0028, \"closest_placement_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia;s__Escherichia flexneri\", \"closest_placement_ani\": 97.95, \"closest_placement_af\": 0.89, \"pplacer_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000026325.1, s__Escherichia coli_D, 96.9551, 97.4, 0.94; GCF_002949675.1, s__Escherichia dysenteriae, 97.0028, 97.2, 0.85; GCF_003697165.2, s__Escherichia coli, 96.8144, 97.0, 0.9; GCA_003018335.1, s__Escherichia coli_C, 96.1333, 96.09, 0.93; GCF_002965065.1, s__Escherichia sp002965065, 95.0, 94.55, 0.77; GCF_004211955.1, s__Escherichia sp004211955, 95.0, 93.11, 0.88; GCF_000208585.1, s__Escherichia sp000208585, 95.0, 92.96, 0.87; GCF_005843885.1, s__Escherichia sp005843885, 95.0, 92.82, 0.89; GCF_011881725.1, s__Escherichia coli_E, 95.0, 92.4, 0.85; GCF_001660175.1, s__Escherichia sp001660175, 95.0, 91.92, 0.88; GCF_000026225.1, s__Escherichia fergusonii, 95.0, 91.19, 0.76; GCF_002900365.1, s__Escherichia marmotae, 95.0, 91.13, 0.86; GCF_000759775.1, s__Escherichia albertii, 95.0, 90.31, 0.77\", \"msa_percent\": 95.81, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.29.fa\", \"classification\": \"Oscillospirales\", \"fastani_reference\": null, \"fastani_reference_radius\": null, \"fastani_taxonomy\": null, \"fastani_ani\": null, \"fastani_af\": null, \"closest_placement_reference\": \"GCA_900540775.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Gemmiger;s__Gemmiger sp900540775\", \"closest_placement_ani\": 86.43, \"closest_placement_af\": 0.75, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Gemmiger;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": null, \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_900167555.1, s__Gemmiger formicilis, 95.0, 87.38, 0.8; GCA_900540595.1, s__Gemmiger sp900540595, 95.0, 86.28, 0.77; GCA_900554145.1, s__Gemmiger sp900554145, 95.0, 84.65, 0.76; GCA_004554775.1, s__Gemmiger variabilis_B, 95.0, 82.89, 0.7; GCA_900539695.1, s__Gemmiger sp900539695, 95.0, 82.69, 0.62; GCA_900772715.1, s__Gemmiger sp900772715, 95.0, 82.44, 0.59; GCA_004552305.1, s__Gemmiger variabilis_A, 95.0, 81.62, 0.63; GCA_900545545.1, s__Gemmiger sp900545545, 95.0, 80.78, 0.5; GCF_003324125.1, s__Gemmiger qucibialis, 95.0, 80.55, 0.45; GCA_900556255.1, s__Gemmiger sp900556255, 95.0, 80.53, 0.65; GCA_003343905.1, s__Gemmiger variabilis_C, 95.0, 80.53, 0.51; GCF_000157955.1, s__Gemmiger variabilis, 95.0, 80.5, 0.49; GCA_004561545.1, s__Gemmiger sp004561545, 95.0, 79.6, 0.43; GCA_002306375.1, s__Gemmiger sp002306375, 95.0, 79.36, 0.43; GCA_900548355.1, s__Gemmiger sp900548355, 95.0, 78.67, 0.34; GCA_004555405.1, s__Gemmiger sp004555405, 95.0, 77.68, 0.17\", \"msa_percent\": 93.59, \"translation_table\": 11.0, \"red_value\": 0.9969970807359956, \"warnings\": \"Genome not assigned to closest species as it falls outside its pre-defined ANI radius\"}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.30.fa\", \"classification\": \"Acidaminococcales\", \"fastani_reference\": \"GCF_900291475.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus provencensis\", \"fastani_ani\": 99.51, \"fastani_af\": 0.97, \"closest_placement_reference\": \"GCF_900291475.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__Acidaminococcus provencensis\", \"closest_placement_ani\": 99.51, \"closest_placement_af\": 0.97, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_C;c__Negativicutes;o__Acidaminococcales;f__Acidaminococcaceae;g__Acidaminococcus;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_009696675.1, s__Acidaminococcus sp000437815, 95.0, 80.19, 0.5; GCA_900770265.1, s__Acidaminococcus sp900770265, 95.0, 80.07, 0.51; GCF_000025305.1, s__Acidaminococcus fermentans, 95.0, 80.06, 0.54; GCF_900106585.1, s__Acidaminococcus timonensis, 95.0, 79.73, 0.5; GCF_900095825.1, s__Acidaminococcus massiliensis, 95.0, 79.6, 0.48; GCA_004555735.1, s__Acidaminococcus fermentans_A, 95.0, 79.22, 0.43; GCA_900314165.1, s__Acidaminococcus sp900314165, 95.0, 79.04, 0.35; GCA_902796945.1, s__Acidaminococcus sp902796945, 95.0, 77.91, 0.22; GCA_900315205.1, s__Acidaminococcus sp900315205, 95.0, 77.7, 0.2; GCA_900538365.1, s__Acidaminococcus sp900538365, 95.0, 77.65, 0.18; GCF_000425045.1, s__Acidaminococcus intestini, 95.0, 77.52, 0.17; GCA_900554515.1, s__Acidaminococcus sp900554515, 95.0, 77.39, 0.27\", \"msa_percent\": 88.11, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.33.fa\", \"classification\": \"Lactobacillales\", \"fastani_reference\": \"GCF_000785515.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Streptococcaceae;g__Streptococcus;s__Streptococcus salivarius\", \"fastani_ani\": 96.08, \"fastani_af\": 0.88, \"closest_placement_reference\": \"GCF_000785515.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Streptococcaceae;g__Streptococcus;s__Streptococcus salivarius\", \"closest_placement_ani\": 96.08, \"closest_placement_af\": 0.88, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactobacillales;f__Streptococcaceae;g__Streptococcus;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_009738225.1, s__Streptococcus sp001556435, 95.0, 94.69, 0.92; GCA_003521145.1, s__Streptococcus sp003521145, 95.0, 93.61, 0.74; GCF_000188295.1, s__Streptococcus vestibularis, 95.0, 92.21, 0.8; GCF_010604095.1, s__Streptococcus sp000187445, 95.0, 89.94, 0.84; GCF_010120595.1, s__Streptococcus thermophilus, 95.0, 89.1, 0.78; GCF_000164675.2, s__Streptococcus parasanguinis, 95.0, 82.13, 0.25; GCF_000448565.1, s__Streptococcus sp000448565, 95.0, 81.99, 0.25; GCF_000963275.1, s__Streptococcus parasanguinis_B, 95.0, 81.91, 0.23; GCF_000180035.1, s__Streptococcus parasanguinis_C, 95.0, 81.83, 0.22; GCF_001073155.1, s__Streptococcus parasanguinis_D, 95.0, 81.8, 0.24; GCF_000186465.1, s__Streptococcus australis, 95.0, 81.64, 0.21; GCF_001074805.1, s__Streptococcus parasanguinis_A, 95.0, 81.44, 0.21; GCF_902363395.1, s__Streptococcus sp902363395, 95.0, 81.17, 0.23; GCF_014435245.1, s__Streptococcus sp900543065, 95.0, 81.0, 0.23; GCF_004166885.1, s__Streptococcus sp004166885, 95.0, 80.83, 0.2; GCF_001587175.1, s__Streptococcus sp001587175, 95.0, 80.83, 0.22; GCF_000314795.2, s__Streptococcus sp000314795, 95.0, 80.67, 0.23; GCF_004785935.1, s__Streptococcus rubneri, 95.0, 80.59, 0.21; GCA_902373455.1, s__Streptococcus sp902373455, 95.0, 80.46, 0.22; GCF_003627135.1, s__Streptococcus koreensis, 95.0, 80.27, 0.21; GCF_004352735.1, s__Streptococcus sp002393675, 95.1145, 79.93, 0.28; GCF_900101445.1, s__Streptococcus equinus_B, 95.0, 79.75, 0.26; GCF_000154985.1, s__Streptococcus infantarius, 95.0, 79.62, 0.28; GCF_900459225.1, s__Streptococcus dysgalactiae, 95.0, 79.55, 0.15; GCA_900755085.1, s__Streptococcus sp900755085, 95.0, 79.48, 0.16; GCF_000186445.1, s__Streptococcus agalactiae, 95.0, 79.41, 0.13; GCF_001553855.1, s__Streptococcus gordonii, 95.0, 79.41, 0.17; GCF_900104225.1, s__Streptococcus equinus_D, 95.1145, 79.38, 0.27; GCF_000960035.1, s__Streptococcus oralis_G, 95.0, 79.38, 0.21; GCF_900478025.1, s__Streptococcus pasteurianus, 96.1304, 79.28, 0.24; GCF_900636575.1, s__Streptococcus canis, 95.0, 79.24, 0.15; GCF_000423745.1, s__Streptococcus plurextorum, 95.0, 79.23, 0.15; GCF_000283635.1, s__Streptococcus macedonicus, 96.3165, 79.21, 0.27; GCF_900459405.1, s__Streptococcus hyointestinalis, 95.0, 79.19, 0.15; GCF_003337175.1, s__Streptococcus gallolyticus_B, 95.6693, 79.18, 0.24; GCF_002953735.1, s__Streptococcus pluranimalium, 95.0, 79.17, 0.18; GCF_000380145.1, s__Streptococcus thoraltensis, 95.0, 79.16, 0.18; GCF_002887775.1, s__Streptococcus penaeicida, 95.0, 79.14, 0.15; GCF_002000985.1, s__Streptococcus gallolyticus, 96.3165, 79.13, 0.24; GCF_000187265.1, s__Streptococcus equinus, 95.0, 79.12, 0.24; GCF_001578885.1, s__Streptococcus sp001578885, 95.0, 79.11, 0.12; GCF_900475675.1, s__Streptococcus lutetiensis, 95.0, 79.05, 0.25; GCF_002964045.1, s__Streptococcus suis_U, 95.0, 79.02, 0.16; GCF_002386345.1, s__Streptococcus oralis_S, 95.0, 79.01, 0.21; GCA_900546335.1, s__Streptococcus sp900546335, 95.0, 79.01, 0.22; GCF_012396585.1, s__Streptococcus ovuberis, 95.0, 78.98, 0.07; GCF_000379985.1, s__Streptococcus caballi, 95.0, 78.97, 0.23; GCF_002055535.1, s__Streptococcus pyogenes, 95.0, 78.96, 0.14; GCF_000376985.1, s__Streptococcus henryi, 95.0, 78.93, 0.27; GCF_012277075.1, s__Streptococcus alactolyticus, 95.0, 78.92, 0.23; GCF_011039275.1, s__Streptococcus hyointestinalis_A, 95.0, 78.9, 0.15; GCF_000425025.1, s__Streptococcus castoreus, 95.0, 78.89, 0.13; GCF_007859195.1, s__Streptococcus pharyngis, 95.0, 78.89, 0.14; GCF_000188055.2, s__Streptococcus urinalis, 95.0, 78.88, 0.13; GCA_900637675.1, s__Streptococcus equi, 95.0, 78.88, 0.12; GCF_900475595.1, s__Streptococcus uberis, 95.0, 78.86, 0.14; GCF_011421455.1, s__Streptococcus sp011421455, 95.0, 78.84, 0.14; GCF_001885095.1, s__Streptococcus bovimastitidis, 95.0, 78.81, 0.16; GCF_003686955.1, s__Streptococcus sp003686955, 95.0, 78.81, 0.19; GCF_001182825.2, s__Streptococcus sp001182825, 95.0, 78.8, 0.16; GCF_000188015.2, s__Streptococcus ictaluri, 95.0, 78.76, 0.16; GCF_000785785.1, s__Streptococcus uberis_A, 95.0, 78.75, 0.13; GCF_007859205.1, s__Streptococcus sp007859205, 95.0, 78.75, 0.17; GCF_900475415.1, s__Streptococcus porcinus, 95.0, 78.74, 0.12; GCF_000420785.1, s__Streptococcus hyovaginalis, 95.0, 78.74, 0.16; GCF_001921845.1, s__Streptococcus cuniculi, 95.0, 78.73, 0.11; GCF_000423765.1, s__Streptococcus porci, 95.0, 78.68, 0.18; GCF_000380005.1, s__Streptococcus didelphis, 95.0, 78.67, 0.11; GCF_000188035.1, s__Streptococcus pseudoporcinus, 95.0, 78.66, 0.13; GCF_001598035.1, s__Streptococcus halotolerans, 95.0, 78.63, 0.17; GCF_000380085.1, s__Streptococcus merionis, 95.0, 78.63, 0.08; GCA_000440235.1, s__Streptococcus suis_F, 95.0, 78.6, 0.14; GCF_001984715.1, s__Streptococcus azizii, 95.0, 78.58, 0.09; GCF_000686605.1, s__Streptococcus sobrinus, 95.0, 78.57, 0.13; GCA_900555155.1, s__Streptococcus sp900555155, 95.0, 78.44, 0.17; GCF_001708305.1, s__Streptococcus himalayensis, 95.0, 78.44, 0.11; GCF_000380105.1, s__Streptococcus orisratti, 95.0, 78.43, 0.2; GCF_901542335.1, s__Streptococcus porcinus_A, 95.0, 78.43, 0.13; GCF_009870755.1, s__Streptococcus halichoeri, 95.0, 78.42, 0.1; GCF_001302265.1, s__Streptococcus phocae, 95.0, 78.4, 0.14; GCF_002760245.1, s__Streptococcus suis_I, 95.0, 78.38, 0.14; GCF_003674745.1, s__Streptococcus iniae, 95.0, 78.37, 0.14; GCF_902729355.1, s__Streptococcus sp902729355, 95.0, 78.35, 0.19; GCF_000187935.1, s__Streptococcus parauberis, 95.0, 78.34, 0.15; GCF_000380025.1, s__Streptococcus entericus, 95.0, 78.3, 0.09; GCF_001431045.1, s__Streptococcus orisasini, 95.0, 78.19, 0.15; GCF_000286075.1, s__Streptococcus ratti, 95.0, 78.16, 0.14; GCF_000423725.1, s__Streptococcus devriesei, 95.0, 78.14, 0.13; GCF_000372425.1, s__Streptococcus ferus, 95.0, 78.08, 0.12; GCF_002355215.1, s__Streptococcus troglodytae, 95.0, 78.03, 0.15; GCF_006739205.1, s__Streptococcus mutans, 95.0, 78.0, 0.16; GCF_000380045.1, s__Streptococcus marimammalium, 95.0, 77.91, 0.1; GCF_001937065.1, s__Streptococcus sp001937065, 95.0, 77.77, 0.14; GCF_000187975.2, s__Streptococcus criceti, 95.0, 77.76, 0.15; GCF_011038795.1, s__Streptococcus sp011038795, 95.0, 77.69, 0.15; GCF_000187995.2, s__Streptococcus macacae, 95.0, 77.65, 0.11; GCF_900459175.1, s__Streptococcus downei, 95.0, 77.52, 0.17; GCF_001642085.1, s__Streptococcus pantholopis, 95.0, 77.48, 0.06; GCF_003086355.1, s__Streptococcus sp003086355, 95.0, 77.44, 0.08\", \"msa_percent\": 94.44, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.34.fa\", \"classification\": \"Oscillospirales\", \"fastani_reference\": \"GCF_010509575.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii_C\", \"fastani_ani\": 97.24, \"fastani_af\": 0.95, \"closest_placement_reference\": \"GCF_010509575.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii_C\", \"closest_placement_ani\": 97.24, \"closest_placement_af\": 0.95, \"pplacer_taxonomy\": \"d__Bacteria;p__Firmicutes_A;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_002550035.1, s__Faecalibacterium prausnitzii_E, 95.0, 86.25, 0.87; GCF_002549975.1, s__Faecalibacterium prausnitzii_H, 95.0, 85.99, 0.88; GCA_900539945.1, s__Faecalibacterium sp900539945, 95.0, 85.3, 0.79; GCF_003287495.1, s__Faecalibacterium prausnitzii_I, 95.0, 85.16, 0.8; GCF_002549775.1, s__Faecalibacterium prausnitzii_F, 95.0, 85.09, 0.83; GCF_003324185.1, s__Faecalibacterium prausnitzii, 95.0612, 84.98, 0.82; GCF_002549755.1, s__Faecalibacterium prausnitzii_D, 95.0, 84.92, 0.82; GCA_900539885.1, s__Faecalibacterium sp900539885, 95.0, 84.86, 0.79; GCF_003287405.1, s__Faecalibacterium prausnitzii_J, 95.0, 84.8, 0.81; GCF_003293635.1, s__Faecalibacterium prausnitzii_G, 95.0, 84.72, 0.81; GCA_900765705.1, s__Faecalibacterium sp900765705, 95.0, 84.61, 0.62; GCF_002550015.1, s__Faecalibacterium prausnitzii_A, 95.0612, 84.53, 0.83; GCA_900772565.1, s__Faecalibacterium sp900772565, 95.0, 84.38, 0.74; GCA_900758465.1, s__Faecalibacterium sp900758465, 95.0, 84.22, 0.77; GCA_003449675.1, s__Faecalibacterium sp003449675, 95.0, 84.06, 0.58; GCA_900551435.1, s__Faecalibacterium sp900551435, 95.0, 83.99, 0.77; GCA_900765105.1, s__Faecalibacterium sp900765105, 95.0, 83.36, 0.67; GCA_004558805.1, s__Faecalibacterium prausnitzii_M, 95.0, 83.27, 0.77; GCA_002313795.1, s__Faecalibacterium prausnitzii_L, 95.0, 83.18, 0.64; GCA_900540455.1, s__Faecalibacterium sp900540455, 95.0, 81.79, 0.62; GCF_002160895.1, s__Faecalibacterium sp002160895, 95.0, 81.44, 0.61; GCF_002160915.1, s__Faecalibacterium sp002160915, 95.0, 80.83, 0.6\", \"msa_percent\": 73.28, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.39.fa\", \"classification\": \"Coriobacteriales\", \"fastani_reference\": null, \"fastani_reference_radius\": null, \"fastani_taxonomy\": null, \"fastani_ani\": null, \"fastani_af\": null, \"closest_placement_reference\": \"GCF_009695875.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Coriobacteriia;o__Coriobacteriales;f__Atopobiaceae;g__Olsenella;s__Olsenella sp900555995\", \"closest_placement_ani\": 92.57, \"closest_placement_af\": 0.9, \"pplacer_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Coriobacteriia;o__Coriobacteriales;f__Atopobiaceae;g__Olsenella;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": null, \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCA_003862195.1, s__Olsenella sp003862195, 95.0, 81.21, 0.5; GCA_902787335.1, s__Olsenella sp902787335, 95.0, 81.13, 0.53; GCA_900313175.1, s__Olsenella sp900313175, 95.0, 81.13, 0.53; GCF_012844235.1, s__Olsenella sp900538935, 95.0, 80.94, 0.51; GCF_900120385.1, s__Olsenella phocaeensis, 95.0, 80.9, 0.45; GCF_900105025.1, s__Olsenella umbonata, 95.0, 80.67, 0.48; GCF_000143845.1, s__Olsenella uli, 95.0, 80.13, 0.44; GCF_900155635.1, s__Olsenella urininfantis, 95.0, 80.09, 0.41; GCF_001457795.1, s__Olsenella sp001457795, 95.0, 79.91, 0.43; GCF_000468755.1, s__Olsenella profusa, 95.0, 79.87, 0.38; GCF_900111255.1, s__Olsenella umbonata_A, 95.0, 79.86, 0.4; GCA_900316105.1, s__Olsenella sp900316105, 95.0, 79.44, 0.32; GCF_001189515.2, s__Olsenella sp001189515, 95.0, 78.06, 0.27\", \"msa_percent\": 94.54, \"translation_table\": 11.0, \"red_value\": 0.994564213248505, \"warnings\": \"Genome not assigned to closest species as it falls outside its pre-defined ANI radius\"}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.42.fa\", \"classification\": \"Actinomycetales\", \"fastani_reference\": \"GCF_000196555.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium longum\", \"fastani_ani\": 99.02, \"fastani_af\": 0.98, \"closest_placement_reference\": \"GCF_000196555.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium longum\", \"closest_placement_ani\": 99.02, \"closest_placement_af\": 0.98, \"pplacer_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000269965.1, s__Bifidobacterium infantis, 95.0, 95.44, 0.9; GCF_001025175.1, s__Bifidobacterium breve, 95.0, 86.49, 0.79; GCF_003129905.1, s__Bifidobacterium callitrichidarum, 95.0, 83.99, 0.8; GCF_002259745.1, s__Bifidobacterium myosotis, 95.0, 83.37, 0.76; GCF_000741695.1, s__Bifidobacterium reuteri, 95.0, 83.0, 0.69; GCF_009193305.1, s__Bifidobacterium sp009193305, 95.0, 82.93, 0.68; GCF_002802915.1, s__Bifidobacterium felsineum, 95.0, 82.7, 0.7; GCF_012932375.1, s__Bifidobacterium sp012932375, 95.0, 82.47, 0.67; GCF_008698235.1, s__Bifidobacterium rousetti, 95.0, 82.41, 0.61; GCF_000741715.1, s__Bifidobacterium saguini, 95.0, 82.34, 0.65; GCF_002802865.1, s__Bifidobacterium scaligerum, 95.0, 82.25, 0.67; GCF_002860405.1, s__Bifidobacterium imperatoris, 95.0, 82.21, 0.66; GCF_002860365.1, s__Bifidobacterium parmae, 95.0, 82.18, 0.61; GCF_010667645.1, s__Bifidobacterium sp010667645, 95.0, 82.09, 0.62; GCF_001042635.1, s__Bifidobacterium scardovii, 95.0, 82.02, 0.66; GCF_000741785.1, s__Bifidobacterium stellenboschense, 95.0, 82.0, 0.64; GCF_000741175.1, s__Bifidobacterium callitrichos, 95.0, 81.96, 0.64; GCF_001417815.1, s__Bifidobacterium aesculapii, 95.0, 81.87, 0.6; GCF_012932365.1, s__Bifidobacterium sp012932365, 95.0, 81.53, 0.52; GCF_009299475.1, s__Bifidobacterium avesanii, 95.0, 81.45, 0.58; GCF_001025135.1, s__Bifidobacterium bifidum, 95.0, 81.44, 0.53; GCF_008698145.1, s__Bifidobacterium vespertilionis, 95.0, 81.33, 0.54; GCF_012932425.1, s__Bifidobacterium sp012932425, 95.0, 81.22, 0.64; GCF_002259685.1, s__Bifidobacterium eulemuris, 95.0, 81.21, 0.6; GCF_001895165.1, s__Bifidobacterium lemurum, 95.0, 81.16, 0.61; GCF_009193355.1, s__Bifidobacterium sp009193355, 95.0, 81.15, 0.55; GCF_012932675.1, s__Bifidobacterium sp012932675, 95.0, 81.13, 0.54; GCF_009299505.1, s__Bifidobacterium ramosum, 95.0, 81.1, 0.61; GCF_010667685.1, s__Bifidobacterium aerophilum, 95.0, 81.07, 0.6; GCF_000741165.1, s__Bifidobacterium biavatii, 95.0, 81.05, 0.59; GCF_900129045.1, s__Bifidobacterium merycicum, 95.0, 80.84, 0.52; GCF_001025155.1, s__Bifidobacterium angulatum, 95.0, 80.82, 0.46; GCF_000770925.1, s__Bifidobacterium ruminantium, 95.0, 80.62, 0.48; GCF_000010425.1, s__Bifidobacterium adolescentis, 95.0, 80.62, 0.53; GCF_003952945.1, s__Bifidobacterium samirii, 95.0, 80.6, 0.57; GCF_000741215.1, s__Bifidobacterium gallinarum, 95.0, 80.58, 0.51; GCF_000771405.1, s__Bifidobacterium pullorum, 95.0, 80.54, 0.51; GCF_009078285.1, s__Bifidobacterium jacchi, 95.0, 80.5, 0.52; GCA_002451435.1, s__Bifidobacterium sp002451435, 95.0, 80.5, 0.48; GCF_001025215.1, s__Bifidobacterium pseudocatenulatum, 95.0, 80.13, 0.47; GCF_001025195.1, s__Bifidobacterium catenulatum, 96.3948, 80.09, 0.41; GCF_001042595.1, s__Bifidobacterium dentium, 95.0, 80.05, 0.46; GCF_001042615.1, s__Bifidobacterium kashiwanohense, 96.3948, 79.99, 0.44; GCF_003951095.1, s__Bifidobacterium goeldii, 95.0, 79.93, 0.37; GCF_000800455.1, s__Bifidobacterium kashiwanohense_A, 95.1942, 79.8, 0.46; GCF_002742445.1, s__Bifidobacterium sp002742445, 95.0, 79.76, 0.43; GCA_002298605.1, s__Bifidobacterium sp002298605, 95.0, 79.7, 0.39; GCF_010667615.1, s__Bifidobacterium sp010667615, 95.0, 79.68, 0.38; GCF_012932445.1, s__Bifidobacterium sp012932445, 95.0, 79.64, 0.36; GCF_000741575.1, s__Bifidobacterium cuniculi, 95.0, 79.61, 0.41; GCF_000522505.1, s__Bifidobacterium moukalabense, 95.0, 79.59, 0.53; GCF_000741295.1, s__Bifidobacterium globosum, 95.0, 79.3, 0.37; GCF_004155535.1, s__Bifidobacterium pseudolongum_C, 95.0, 79.24, 0.39; GCF_000741135.1, s__Bifidobacterium choerinum, 95.0, 79.15, 0.39; GCF_000771265.1, s__Bifidobacterium thermophilum, 95.0, 79.1, 0.38; GCF_000771225.1, s__Bifidobacterium pseudolongum, 95.0, 78.99, 0.37; GCA_000741495.1, s__Bifidobacterium thermophilum_A, 95.0, 78.96, 0.36; GCF_002860345.1, s__Bifidobacterium anseris, 95.0, 78.94, 0.37; GCF_003952025.1, s__Bifidobacterium castoris, 95.0, 78.94, 0.39; GCF_000741535.1, s__Bifidobacterium boum, 95.0, 78.92, 0.35; GCF_002286915.1, s__Bifidobacterium italicum, 95.0, 78.88, 0.38; GCF_000741255.1, s__Bifidobacterium magnum, 95.0, 78.74, 0.28; GCF_009299485.1, s__Bifidobacterium apri, 95.0, 78.74, 0.34; GCF_002259755.1, s__Bifidobacterium hapali, 95.0, 78.69, 0.3; GCF_000741775.1, s__Bifidobacterium subtile, 95.0, 78.61, 0.41; GCF_009371885.1, s__Bifidobacterium tibiigranuli, 95.0, 78.58, 0.38; GCF_000741285.1, s__Bifidobacterium mongoliense, 95.0, 78.47, 0.32; GCF_000260715.1, s__Bifidobacterium animalis, 95.0, 78.36, 0.35; GCF_009727065.1, s__Bifidobacterium sp900551485, 95.0, 78.06, 0.33; GCF_002286935.1, s__Bifidobacterium criceti, 95.0, 77.98, 0.3; GCF_001263395.1, s__Bifidobacterium actinocoloniiforme, 95.0, 77.93, 0.18; GCF_003315635.1, s__Bifidobacterium aemilianum, 95.0, 77.93, 0.28; GCF_000741205.1, s__Bifidobacterium gallicum, 95.0, 77.89, 0.23; GCF_000706765.1, s__Bifidobacterium indicum, 95.0, 77.82, 0.15; GCF_000741645.1, s__Bifidobacterium minimum, 95.0, 77.79, 0.23; GCF_003202755.1, s__Bifidobacterium asteroides_F, 95.0, 77.7, 0.2; GCF_003202715.1, s__Bifidobacterium asteroides_E, 95.0, 77.68, 0.2; GCF_000741525.1, s__Bifidobacterium bohemicum, 95.0, 77.65, 0.24; GCF_000263635.1, s__Bifidobacterium vaginale_C, 95.0, 77.42, 0.05; GCF_003315615.1, s__Bifidobacterium xylocopae, 95.0, 77.4, 0.2; GCF_000741765.1, s__Bifidobacterium tsurumiense, 95.0, 77.36, 0.19; GCF_000738005.1, s__Bifidobacterium crudilactis, 95.0, 77.36, 0.2; GCF_000741705.1, s__Bifidobacterium psychraerophilum, 95.0, 77.22, 0.21; GCF_003951975.1, s__Bifidobacterium dolichotidis, 95.0, 77.2, 0.14; GCF_900094885.1, s__Bifidobacterium commune, 95.0, 77.18, 0.14; GCF_001546485.1, s__Bifidobacterium vaginale_D, 95.0, 77.12, 0.05; GCF_003202695.1, s__Bifidobacterium asteroides_G, 95.0, 77.11, 0.18; GCF_000263595.1, s__Bifidobacterium vaginale_E, 95.0, 77.06, 0.05; GCF_003408845.1, s__Bifidobacterium vaginale_H, 95.0, 76.97, 0.05; GCF_003585845.1, s__Bifidobacterium sp003585845, 95.0, 76.95, 0.05; GCF_003585735.1, s__Bifidobacterium sp003585735, 95.0, 76.77, 0.06; GCF_001563665.1, s__Bifidobacterium vaginale_A, 95.0, 76.75, 0.03; GCF_000737845.1, s__Bifidobacterium bombi, 95.0, 76.62, 0.13; GCF_002896555.1, s__Bifidobacterium vaginale_F, 95.0, 76.59, 0.05; GCF_000025205.1, s__Bifidobacterium vaginale_G, 95.0, 76.54, 0.05; GCF_002259795.1, s__Bifidobacterium aquikefiri, 95.0, 76.42, 0.08; GCF_001546455.1, s__Bifidobacterium vaginale_B, 95.0, 76.35, 0.05; GCF_001042655.1, s__Bifidobacterium vaginale, 95.0, 76.31, 0.04; GCF_002884815.1, s__Bifidobacterium sp002884815, 95.0, 76.3, 0.05\", \"msa_percent\": 78.26, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.45.fa\", \"classification\": \"Actinomycetales\", \"fastani_reference\": \"GCF_000010425.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium adolescentis\", \"fastani_ani\": 98.41, \"fastani_af\": 0.92, \"closest_placement_reference\": \"GCF_000010425.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__Bifidobacterium adolescentis\", \"closest_placement_ani\": 98.41, \"closest_placement_af\": 0.92, \"pplacer_taxonomy\": \"d__Bacteria;p__Actinobacteriota;c__Actinomycetia;o__Actinomycetales;f__Bifidobacteriaceae;g__Bifidobacterium;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_000770925.1, s__Bifidobacterium ruminantium, 95.0, 88.72, 0.82; GCF_001025195.1, s__Bifidobacterium catenulatum, 96.3948, 84.96, 0.77; GCF_001042615.1, s__Bifidobacterium kashiwanohense, 96.3948, 84.43, 0.77; GCF_001025215.1, s__Bifidobacterium pseudocatenulatum, 95.0, 84.22, 0.77; GCF_002742445.1, s__Bifidobacterium sp002742445, 95.0, 84.19, 0.74; GCA_002451435.1, s__Bifidobacterium sp002451435, 95.0, 84.11, 0.66; GCF_000800455.1, s__Bifidobacterium kashiwanohense_A, 95.1942, 83.86, 0.74; GCF_000522505.1, s__Bifidobacterium moukalabense, 95.0, 83.57, 0.76; GCF_001042595.1, s__Bifidobacterium dentium, 95.0, 83.28, 0.71; GCF_012932675.1, s__Bifidobacterium sp012932675, 95.0, 82.89, 0.63; GCF_900129045.1, s__Bifidobacterium merycicum, 95.0, 82.7, 0.63; GCF_001025155.1, s__Bifidobacterium angulatum, 95.0, 82.68, 0.59; GCF_010667645.1, s__Bifidobacterium sp010667645, 95.0, 81.39, 0.59; GCF_003129905.1, s__Bifidobacterium callitrichidarum, 95.0, 81.34, 0.56; GCF_002860365.1, s__Bifidobacterium parmae, 95.0, 81.3, 0.58; GCF_000269965.1, s__Bifidobacterium infantis, 95.0, 81.16, 0.54; GCF_000741785.1, s__Bifidobacterium stellenboschense, 95.0, 81.12, 0.58; GCF_001895165.1, s__Bifidobacterium lemurum, 95.0, 81.11, 0.57; GCF_010667685.1, s__Bifidobacterium aerophilum, 95.0, 81.1, 0.59; GCF_009299475.1, s__Bifidobacterium avesanii, 95.0, 81.09, 0.52; GCF_009299505.1, s__Bifidobacterium ramosum, 95.0, 81.04, 0.61; GCF_001042635.1, s__Bifidobacterium scardovii, 95.0, 80.96, 0.6; GCF_003952945.1, s__Bifidobacterium samirii, 95.0, 80.95, 0.56; GCF_002259685.1, s__Bifidobacterium eulemuris, 95.0, 80.92, 0.59; GCF_000741165.1, s__Bifidobacterium biavatii, 95.0, 80.91, 0.58; GCF_001025135.1, s__Bifidobacterium bifidum, 95.0, 80.86, 0.53; GCF_000741175.1, s__Bifidobacterium callitrichos, 95.0, 80.82, 0.58; GCF_009193305.1, s__Bifidobacterium sp009193305, 95.0, 80.79, 0.48; GCF_000771405.1, s__Bifidobacterium pullorum, 95.0, 80.73, 0.51; GCF_008698235.1, s__Bifidobacterium rousetti, 95.0, 80.71, 0.57; GCF_000196555.1, s__Bifidobacterium longum, 95.0, 80.71, 0.53; GCF_012932365.1, s__Bifidobacterium sp012932365, 95.0, 80.69, 0.52; GCF_002259745.1, s__Bifidobacterium myosotis, 95.0, 80.68, 0.57; GCF_000741215.1, s__Bifidobacterium gallinarum, 95.0, 80.66, 0.51; GCF_009193355.1, s__Bifidobacterium sp009193355, 95.0, 80.62, 0.57; GCF_008698145.1, s__Bifidobacterium vespertilionis, 95.0, 80.61, 0.5; GCF_001025175.1, s__Bifidobacterium breve, 95.0, 80.59, 0.48; GCF_001417815.1, s__Bifidobacterium aesculapii, 95.0, 80.54, 0.57; GCF_000741695.1, s__Bifidobacterium reuteri, 95.0, 80.54, 0.51; GCF_002802915.1, s__Bifidobacterium felsineum, 95.0, 80.47, 0.47; GCF_002860405.1, s__Bifidobacterium imperatoris, 95.0, 80.42, 0.41; GCF_012932375.1, s__Bifidobacterium sp012932375, 95.0, 80.39, 0.47; GCF_002802865.1, s__Bifidobacterium scaligerum, 95.0, 80.33, 0.47; GCF_000741715.1, s__Bifidobacterium saguini, 95.0, 80.12, 0.44; GCF_012932425.1, s__Bifidobacterium sp012932425, 95.0, 80.12, 0.5; GCF_012932445.1, s__Bifidobacterium sp012932445, 95.0, 80.1, 0.35; GCF_009078285.1, s__Bifidobacterium jacchi, 95.0, 80.07, 0.48; GCF_000741575.1, s__Bifidobacterium cuniculi, 95.0, 80.04, 0.42; GCF_003951095.1, s__Bifidobacterium goeldii, 95.0, 79.98, 0.37; GCA_002298605.1, s__Bifidobacterium sp002298605, 95.0, 79.84, 0.39; GCF_010667615.1, s__Bifidobacterium sp010667615, 95.0, 79.78, 0.45; GCF_000741135.1, s__Bifidobacterium choerinum, 95.0, 79.76, 0.47; GCA_000741495.1, s__Bifidobacterium thermophilum_A, 95.0, 79.76, 0.43; GCF_000741255.1, s__Bifidobacterium magnum, 95.0, 79.74, 0.34; GCF_000741535.1, s__Bifidobacterium boum, 95.0, 79.73, 0.39; GCF_004155535.1, s__Bifidobacterium pseudolongum_C, 95.0, 79.73, 0.44; GCF_000771265.1, s__Bifidobacterium thermophilum, 95.0, 79.66, 0.42; GCF_000771225.1, s__Bifidobacterium pseudolongum, 95.0, 79.62, 0.41; GCF_002286915.1, s__Bifidobacterium italicum, 95.0, 79.61, 0.44; GCF_000741295.1, s__Bifidobacterium globosum, 95.0, 79.57, 0.42; GCF_009299485.1, s__Bifidobacterium apri, 95.0, 79.51, 0.4; GCF_003952025.1, s__Bifidobacterium castoris, 95.0, 79.51, 0.46; GCF_002860345.1, s__Bifidobacterium anseris, 95.0, 79.32, 0.46; GCF_000260715.1, s__Bifidobacterium animalis, 95.0, 79.16, 0.39; GCF_009727065.1, s__Bifidobacterium sp900551485, 95.0, 79.1, 0.36; GCF_009371885.1, s__Bifidobacterium tibiigranuli, 95.0, 79.01, 0.41; GCF_000741285.1, s__Bifidobacterium mongoliense, 95.0, 78.95, 0.31; GCF_000741775.1, s__Bifidobacterium subtile, 95.0, 78.88, 0.42; GCF_002259755.1, s__Bifidobacterium hapali, 95.0, 78.88, 0.32; GCF_002286935.1, s__Bifidobacterium criceti, 95.0, 78.82, 0.37; GCF_000741205.1, s__Bifidobacterium gallicum, 95.0, 78.56, 0.27; GCF_003315635.1, s__Bifidobacterium aemilianum, 95.0, 78.4, 0.26; GCF_000741765.1, s__Bifidobacterium tsurumiense, 95.0, 78.37, 0.19; GCF_000741525.1, s__Bifidobacterium bohemicum, 95.0, 78.28, 0.22; GCF_003408845.1, s__Bifidobacterium vaginale_H, 95.0, 78.2, 0.06; GCF_000741645.1, s__Bifidobacterium minimum, 95.0, 78.12, 0.23; GCF_900094885.1, s__Bifidobacterium commune, 95.0, 78.06, 0.15; GCF_003315615.1, s__Bifidobacterium xylocopae, 95.0, 78.04, 0.17; GCF_003951975.1, s__Bifidobacterium dolichotidis, 95.0, 78.02, 0.14; GCF_001042655.1, s__Bifidobacterium vaginale, 95.0, 78.01, 0.05; GCF_001263395.1, s__Bifidobacterium actinocoloniiforme, 95.0, 78.0, 0.18; GCF_003202755.1, s__Bifidobacterium asteroides_F, 95.0, 77.99, 0.17; GCF_000706765.1, s__Bifidobacterium indicum, 95.0, 77.88, 0.15; GCF_003202715.1, s__Bifidobacterium asteroides_E, 95.0, 77.84, 0.17; GCF_003585845.1, s__Bifidobacterium sp003585845, 95.0, 77.79, 0.06; GCF_001546485.1, s__Bifidobacterium vaginale_D, 95.0, 77.66, 0.06; GCF_003202695.1, s__Bifidobacterium asteroides_G, 95.0, 77.63, 0.15; GCF_000263635.1, s__Bifidobacterium vaginale_C, 95.0, 77.62, 0.05; GCF_002896555.1, s__Bifidobacterium vaginale_F, 95.0, 77.57, 0.06; GCF_000263595.1, s__Bifidobacterium vaginale_E, 95.0, 77.56, 0.06; GCF_000738005.1, s__Bifidobacterium crudilactis, 95.0, 77.49, 0.21; GCF_003585735.1, s__Bifidobacterium sp003585735, 95.0, 77.47, 0.08; GCF_000025205.1, s__Bifidobacterium vaginale_G, 95.0, 77.39, 0.06; GCF_001563665.1, s__Bifidobacterium vaginale_A, 95.0, 77.26, 0.04; GCF_002884815.1, s__Bifidobacterium sp002884815, 95.0, 77.21, 0.05; GCF_000737845.1, s__Bifidobacterium bombi, 95.0, 77.13, 0.14; GCF_001546455.1, s__Bifidobacterium vaginale_B, 95.0, 77.12, 0.05; GCF_000741705.1, s__Bifidobacterium psychraerophilum, 95.0, 77.11, 0.22; GCF_002259795.1, s__Bifidobacterium aquikefiri, 95.0, 77.04, 0.1\", \"msa_percent\": 82.69, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}, {\"user_genome\": \"MEGAHIT-MetaBAT2-CAPES_S11.5.fa\", \"classification\": \"Burkholderiales\", \"fastani_reference\": \"GCF_003315195.1\", \"fastani_reference_radius\": 95.0, \"fastani_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae;g__Sutterella;s__Sutterella wadsworthensis\", \"fastani_ani\": 97.67, \"fastani_af\": 0.91, \"closest_placement_reference\": \"GCF_003315195.1\", \"closest_placement_radius\": 95.0, \"closest_placement_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae;g__Sutterella;s__Sutterella wadsworthensis\", \"closest_placement_ani\": 97.67, \"closest_placement_af\": 0.91, \"pplacer_taxonomy\": \"d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Burkholderiaceae;g__Sutterella;s__\", \"classification_method\": \"taxonomic classification defined by topology and ANI\", \"note\": \"topological placement and ANI have congruent species assignments\", \"other_related_references(genome_id,species_name,radius,ANI,AF)\": \"GCF_009183815.1, s__Sutterella seckii_A, 95.0, 79.26, 0.32; GCF_006337085.1, s__Sutterella faecalis, 95.0, 78.99, 0.31; GCA_900770335.1, s__Sutterella sp900770335, 95.0, 78.59, 0.28; GCF_000250875.1, s__Sutterella parvirubra, 95.0, 78.42, 0.17; GCF_003609995.1, s__Sutterella megalosphaeroides, 95.0, 78.22, 0.17; GCA_900754475.1, s__Sutterella sp900754475, 95.0, 78.04, 0.19; GCF_000297775.1, s__Sutterella wadsworthensis_A, 95.0, 77.91, 0.15; GCA_900757865.1, s__Sutterella sp900757865, 95.0, 77.91, 0.06; GCA_900543805.1, s__Sutterella sp900543805, 95.0, 77.83, 0.18; GCA_900764215.1, s__Sutterella sp900764215, 95.0, 77.79, 0.17; GCA_900545275.1, s__Sutterella sp900545275, 95.0, 77.76, 0.15; GCA_900762445.1, s__Sutterella sp900762445, 95.0, 77.12, 0.08\", \"msa_percent\": 96.96, \"translation_table\": 11.0, \"red_value\": null, \"warnings\": null}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "execution_count": 489,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alt.Chart(gtdb).mark_bar().encode(\n",
    " alt.X('count(classification):Q'),\n",
    " alt.Y('classification:N', sort='-x')\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "417956e6-5e5c-4878-bd67-0d5b9c1968f0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "a5040315-e5cf-4bad-9c51-deb3a9942365",
   "metadata": {},
   "source": [
    "### Centrifuge classificaiton (right after preprocessing)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3eb7b1bd-1842-4071-8df4-f266e73f13ee",
   "metadata": {},
   "source": [
    "### needs a way to identify which one is virus. Maybe this is sorted out if one uses another db to classify?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "7762e693-12c3-4306-8d65-f1200e63e8e0",
   "metadata": {},
   "outputs": [],
   "source": [
    "centrifuge = pd.read_csv('/Users/williamrosenbaum/Downloads/report.txt',\n",
    "                         sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "f6cfddf2-84a2-4405-abe5-b76099551e27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>taxID</th>\n",
       "      <th>taxRank</th>\n",
       "      <th>genomeSize</th>\n",
       "      <th>numReads</th>\n",
       "      <th>numUniqueReads</th>\n",
       "      <th>abundance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>Klebsiella pneumoniae</td>\n",
       "      <td>573</td>\n",
       "      <td>species</td>\n",
       "      <td>15713353</td>\n",
       "      <td>4403776</td>\n",
       "      <td>3921236</td>\n",
       "      <td>2.082890e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>562</td>\n",
       "      <td>species</td>\n",
       "      <td>12319210</td>\n",
       "      <td>2998448</td>\n",
       "      <td>1302439</td>\n",
       "      <td>2.055870e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1913</th>\n",
       "      <td>Bifidobacterium longum</td>\n",
       "      <td>216816</td>\n",
       "      <td>species</td>\n",
       "      <td>3655960</td>\n",
       "      <td>735601</td>\n",
       "      <td>695952</td>\n",
       "      <td>1.486330e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1513</th>\n",
       "      <td>Streptococcus infantarius</td>\n",
       "      <td>102684</td>\n",
       "      <td>species</td>\n",
       "      <td>1913271</td>\n",
       "      <td>309029</td>\n",
       "      <td>264686</td>\n",
       "      <td>7.275100e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>818</td>\n",
       "      <td>species</td>\n",
       "      <td>6390542</td>\n",
       "      <td>977349</td>\n",
       "      <td>942527</td>\n",
       "      <td>7.028640e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2420</th>\n",
       "      <td>Dehalogenimonas lykanthroporepellens</td>\n",
       "      <td>552810</td>\n",
       "      <td>species</td>\n",
       "      <td>1686510</td>\n",
       "      <td>77</td>\n",
       "      <td>49</td>\n",
       "      <td>1.403740e-177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1983</th>\n",
       "      <td>Fictibacillus arsenicus</td>\n",
       "      <td>255247</td>\n",
       "      <td>species</td>\n",
       "      <td>4055461</td>\n",
       "      <td>49</td>\n",
       "      <td>39</td>\n",
       "      <td>1.128970e-188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2824</th>\n",
       "      <td>Fictibacillus phosphorivorans</td>\n",
       "      <td>1221500</td>\n",
       "      <td>species</td>\n",
       "      <td>4230665</td>\n",
       "      <td>95</td>\n",
       "      <td>73</td>\n",
       "      <td>1.057440e-189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1451</th>\n",
       "      <td>Bacillus weihenstephanensis</td>\n",
       "      <td>86662</td>\n",
       "      <td>species</td>\n",
       "      <td>5740546</td>\n",
       "      <td>135</td>\n",
       "      <td>42</td>\n",
       "      <td>1.123280e-204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1338</th>\n",
       "      <td>Denitrobacterium detoxificans</td>\n",
       "      <td>79604</td>\n",
       "      <td>species</td>\n",
       "      <td>2446250</td>\n",
       "      <td>34</td>\n",
       "      <td>6</td>\n",
       "      <td>6.577480e-220</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>450 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      name    taxID  taxRank  genomeSize  \\\n",
       "186                  Klebsiella pneumoniae      573  species    15713353   \n",
       "179                       Escherichia coli      562  species    12319210   \n",
       "1913                Bifidobacterium longum   216816  species     3655960   \n",
       "1513             Streptococcus infantarius   102684  species     1913271   \n",
       "269           Bacteroides thetaiotaomicron      818  species     6390542   \n",
       "...                                    ...      ...      ...         ...   \n",
       "2420  Dehalogenimonas lykanthroporepellens   552810  species     1686510   \n",
       "1983               Fictibacillus arsenicus   255247  species     4055461   \n",
       "2824         Fictibacillus phosphorivorans  1221500  species     4230665   \n",
       "1451           Bacillus weihenstephanensis    86662  species     5740546   \n",
       "1338         Denitrobacterium detoxificans    79604  species     2446250   \n",
       "\n",
       "      numReads  numUniqueReads      abundance  \n",
       "186    4403776         3921236   2.082890e-01  \n",
       "179    2998448         1302439   2.055870e-01  \n",
       "1913    735601          695952   1.486330e-01  \n",
       "1513    309029          264686   7.275100e-02  \n",
       "269     977349          942527   7.028640e-02  \n",
       "...        ...             ...            ...  \n",
       "2420        77              49  1.403740e-177  \n",
       "1983        49              39  1.128970e-188  \n",
       "2824        95              73  1.057440e-189  \n",
       "1451       135              42  1.123280e-204  \n",
       "1338        34               6  6.577480e-220  \n",
       "\n",
       "[450 rows x 7 columns]"
      ]
     },
     "execution_count": 406,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(centrifuge\n",
    " .loc[lambda x: x['abundance'] > 0]\n",
    " .sort_values(by='abundance', ascending=False)\n",
    "# .loc[lambda x: x['name'].str.contains(r'virus|phage')] #try to filter out virus \n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b177e08-8b5a-4401-a609-f18dbe31ddf3",
   "metadata": {},
   "source": [
    "## Genome binning summary (/results/binning/summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "id": "98440452-e389-4f1c-b799-4717242a6034",
   "metadata": {},
   "outputs": [],
   "source": [
    "binning = pd.read_csv('bin_summary.tsv',\n",
    "                      sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "b483a12d-e1ad-4375-b7d5-e5e3f6c219a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin</th>\n",
       "      <th>Depth CAPES_S11</th>\n",
       "      <th>Depth CAPES_S21</th>\n",
       "      <th>Depth CAPES_S7</th>\n",
       "      <th>GenomeBin</th>\n",
       "      <th>Domain</th>\n",
       "      <th>%Complete (domain)</th>\n",
       "      <th>%Complete and single-copy (domain)</th>\n",
       "      <th>%Complete and duplicated (domain)</th>\n",
       "      <th>%Fragmented (domain)</th>\n",
       "      <th>...</th>\n",
       "      <th>closest_placement_ani</th>\n",
       "      <th>closest_placement_af</th>\n",
       "      <th>pplacer_taxonomy</th>\n",
       "      <th>classification_method</th>\n",
       "      <th>note</th>\n",
       "      <th>other_related_references(genome_id,species_name,radius,ANI,AF)</th>\n",
       "      <th>msa_percent</th>\n",
       "      <th>translation_table</th>\n",
       "      <th>red_value</th>\n",
       "      <th>warnings</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.002.fa</td>\n",
       "      <td>12.13450</td>\n",
       "      <td>165.35450</td>\n",
       "      <td>65.374600</td>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.002.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>92.7</td>\n",
       "      <td>91.1</td>\n",
       "      <td>1.6</td>\n",
       "      <td>0.8</td>\n",
       "      <td>...</td>\n",
       "      <td>98.73</td>\n",
       "      <td>0.89</td>\n",
       "      <td>d__Bacteria;p__Actinobacteriota;c__Actinomycet...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCF_000269965.1, s__Bifidobacterium infantis, ...</td>\n",
       "      <td>90.27</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.003.fa</td>\n",
       "      <td>23.78450</td>\n",
       "      <td>77.88510</td>\n",
       "      <td>77.397200</td>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.003.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>97.6</td>\n",
       "      <td>83.1</td>\n",
       "      <td>14.5</td>\n",
       "      <td>1.6</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.005.fa</td>\n",
       "      <td>1.24615</td>\n",
       "      <td>3.89888</td>\n",
       "      <td>123.401000</td>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.005.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>58.8</td>\n",
       "      <td>54.8</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.4</td>\n",
       "      <td>...</td>\n",
       "      <td>99.03</td>\n",
       "      <td>0.88</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCF_000828055.2, s__Klebsiella variicola, 95.0...</td>\n",
       "      <td>62.16</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.001.fa</td>\n",
       "      <td>4.20057</td>\n",
       "      <td>113.81500</td>\n",
       "      <td>122.391000</td>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.001.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>90.3</td>\n",
       "      <td>77.4</td>\n",
       "      <td>12.9</td>\n",
       "      <td>2.4</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.004.fa</td>\n",
       "      <td>48.09430</td>\n",
       "      <td>4.50323</td>\n",
       "      <td>115.534000</td>\n",
       "      <td>MEGAHIT-MaxBin2-CAPES_S7.004.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>63.7</td>\n",
       "      <td>52.4</td>\n",
       "      <td>11.3</td>\n",
       "      <td>10.5</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.008.fa</td>\n",
       "      <td>1.16163</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>29.996900</td>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.008.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>95.1</td>\n",
       "      <td>91.1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.6</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>d__Bacteria;p__Firmicutes_C;c__Negativicutes;o...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCF_902810435.1, s__Veillonella parvula_A, 95....</td>\n",
       "      <td>88.90</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.996764</td>\n",
       "      <td>Genome not assigned to closest species as it f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.010.fa</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>13.498000</td>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.010.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>42.7</td>\n",
       "      <td>41.9</td>\n",
       "      <td>0.8</td>\n",
       "      <td>8.9</td>\n",
       "      <td>...</td>\n",
       "      <td>95.44</td>\n",
       "      <td>0.59</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...</td>\n",
       "      <td>ANI</td>\n",
       "      <td>topological placement and ANI have incongruent...</td>\n",
       "      <td>GCF_900478025.1, s__Streptococcus pasteurianus...</td>\n",
       "      <td>50.94</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.011.fa</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>10.866700</td>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.011.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>26.6</td>\n",
       "      <td>24.2</td>\n",
       "      <td>2.4</td>\n",
       "      <td>8.9</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.012.fa</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>8.468365</td>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.012.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>76.6</td>\n",
       "      <td>75.0</td>\n",
       "      <td>1.6</td>\n",
       "      <td>11.3</td>\n",
       "      <td>...</td>\n",
       "      <td>98.64</td>\n",
       "      <td>0.71</td>\n",
       "      <td>d__Bacteria;p__Firmicutes_A;c__Clostridia;o__L...</td>\n",
       "      <td>taxonomic classification defined by topology a...</td>\n",
       "      <td>topological placement and ANI have congruent s...</td>\n",
       "      <td>GCA_900546325.1, s__Faecalimonas sp900546325, ...</td>\n",
       "      <td>71.71</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.013.fa</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>8.588700</td>\n",
       "      <td>SPAdes-MaxBin2-CAPES_S7.013.fa</td>\n",
       "      <td>bacteria_odb10</td>\n",
       "      <td>17.7</td>\n",
       "      <td>15.3</td>\n",
       "      <td>2.4</td>\n",
       "      <td>9.7</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>583 rows Ã— 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 bin  Depth CAPES_S11  Depth CAPES_S21  \\\n",
       "0    MEGAHIT-MaxBin2-CAPES_S7.002.fa         12.13450        165.35450   \n",
       "1    MEGAHIT-MaxBin2-CAPES_S7.003.fa         23.78450         77.88510   \n",
       "2    MEGAHIT-MaxBin2-CAPES_S7.005.fa          1.24615          3.89888   \n",
       "3    MEGAHIT-MaxBin2-CAPES_S7.001.fa          4.20057        113.81500   \n",
       "4    MEGAHIT-MaxBin2-CAPES_S7.004.fa         48.09430          4.50323   \n",
       "..                               ...              ...              ...   \n",
       "578   SPAdes-MaxBin2-CAPES_S7.008.fa          1.16163          0.00000   \n",
       "579   SPAdes-MaxBin2-CAPES_S7.010.fa          0.00000          0.00000   \n",
       "580   SPAdes-MaxBin2-CAPES_S7.011.fa          0.00000          0.00000   \n",
       "581   SPAdes-MaxBin2-CAPES_S7.012.fa          0.00000          0.00000   \n",
       "582   SPAdes-MaxBin2-CAPES_S7.013.fa          0.00000          0.00000   \n",
       "\n",
       "     Depth CAPES_S7                        GenomeBin          Domain  \\\n",
       "0         65.374600  MEGAHIT-MaxBin2-CAPES_S7.002.fa  bacteria_odb10   \n",
       "1         77.397200  MEGAHIT-MaxBin2-CAPES_S7.003.fa  bacteria_odb10   \n",
       "2        123.401000  MEGAHIT-MaxBin2-CAPES_S7.005.fa  bacteria_odb10   \n",
       "3        122.391000  MEGAHIT-MaxBin2-CAPES_S7.001.fa  bacteria_odb10   \n",
       "4        115.534000  MEGAHIT-MaxBin2-CAPES_S7.004.fa  bacteria_odb10   \n",
       "..              ...                              ...             ...   \n",
       "578       29.996900   SPAdes-MaxBin2-CAPES_S7.008.fa  bacteria_odb10   \n",
       "579       13.498000   SPAdes-MaxBin2-CAPES_S7.010.fa  bacteria_odb10   \n",
       "580       10.866700   SPAdes-MaxBin2-CAPES_S7.011.fa  bacteria_odb10   \n",
       "581        8.468365   SPAdes-MaxBin2-CAPES_S7.012.fa  bacteria_odb10   \n",
       "582        8.588700   SPAdes-MaxBin2-CAPES_S7.013.fa  bacteria_odb10   \n",
       "\n",
       "     %Complete (domain)  %Complete and single-copy (domain)  \\\n",
       "0                  92.7                                91.1   \n",
       "1                  97.6                                83.1   \n",
       "2                  58.8                                54.8   \n",
       "3                  90.3                                77.4   \n",
       "4                  63.7                                52.4   \n",
       "..                  ...                                 ...   \n",
       "578                95.1                                91.1   \n",
       "579                42.7                                41.9   \n",
       "580                26.6                                24.2   \n",
       "581                76.6                                75.0   \n",
       "582                17.7                                15.3   \n",
       "\n",
       "     %Complete and duplicated (domain)  %Fragmented (domain)  ...  \\\n",
       "0                                  1.6                   0.8  ...   \n",
       "1                                 14.5                   1.6  ...   \n",
       "2                                  4.0                   2.4  ...   \n",
       "3                                 12.9                   2.4  ...   \n",
       "4                                 11.3                  10.5  ...   \n",
       "..                                 ...                   ...  ...   \n",
       "578                                4.0                   1.6  ...   \n",
       "579                                0.8                   8.9  ...   \n",
       "580                                2.4                   8.9  ...   \n",
       "581                                1.6                  11.3  ...   \n",
       "582                                2.4                   9.7  ...   \n",
       "\n",
       "     closest_placement_ani  closest_placement_af  \\\n",
       "0                    98.73                  0.89   \n",
       "1                      NaN                   NaN   \n",
       "2                    99.03                  0.88   \n",
       "3                      NaN                   NaN   \n",
       "4                      NaN                   NaN   \n",
       "..                     ...                   ...   \n",
       "578                    NaN                   NaN   \n",
       "579                  95.44                  0.59   \n",
       "580                    NaN                   NaN   \n",
       "581                  98.64                  0.71   \n",
       "582                    NaN                   NaN   \n",
       "\n",
       "                                      pplacer_taxonomy  \\\n",
       "0    d__Bacteria;p__Actinobacteriota;c__Actinomycet...   \n",
       "1                                                  NaN   \n",
       "2    d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "3                                                  NaN   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "578  d__Bacteria;p__Firmicutes_C;c__Negativicutes;o...   \n",
       "579  d__Bacteria;p__Firmicutes;c__Bacilli;o__Lactob...   \n",
       "580                                                NaN   \n",
       "581  d__Bacteria;p__Firmicutes_A;c__Clostridia;o__L...   \n",
       "582                                                NaN   \n",
       "\n",
       "                                 classification_method  \\\n",
       "0    taxonomic classification defined by topology a...   \n",
       "1                                                  NaN   \n",
       "2    taxonomic classification defined by topology a...   \n",
       "3                                                  NaN   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "578  taxonomic classification defined by topology a...   \n",
       "579                                                ANI   \n",
       "580                                                NaN   \n",
       "581  taxonomic classification defined by topology a...   \n",
       "582                                                NaN   \n",
       "\n",
       "                                                  note  \\\n",
       "0    topological placement and ANI have congruent s...   \n",
       "1                                                  NaN   \n",
       "2    topological placement and ANI have congruent s...   \n",
       "3                                                  NaN   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "578                                                NaN   \n",
       "579  topological placement and ANI have incongruent...   \n",
       "580                                                NaN   \n",
       "581  topological placement and ANI have congruent s...   \n",
       "582                                                NaN   \n",
       "\n",
       "     other_related_references(genome_id,species_name,radius,ANI,AF)  \\\n",
       "0    GCF_000269965.1, s__Bifidobacterium infantis, ...                \n",
       "1                                                  NaN                \n",
       "2    GCF_000828055.2, s__Klebsiella variicola, 95.0...                \n",
       "3                                                  NaN                \n",
       "4                                                  NaN                \n",
       "..                                                 ...                \n",
       "578  GCF_902810435.1, s__Veillonella parvula_A, 95....                \n",
       "579  GCF_900478025.1, s__Streptococcus pasteurianus...                \n",
       "580                                                NaN                \n",
       "581  GCA_900546325.1, s__Faecalimonas sp900546325, ...                \n",
       "582                                                NaN                \n",
       "\n",
       "     msa_percent  translation_table  red_value  \\\n",
       "0          90.27               11.0        NaN   \n",
       "1            NaN                NaN        NaN   \n",
       "2          62.16               11.0        NaN   \n",
       "3            NaN                NaN        NaN   \n",
       "4            NaN                NaN        NaN   \n",
       "..           ...                ...        ...   \n",
       "578        88.90               11.0   0.996764   \n",
       "579        50.94               11.0        NaN   \n",
       "580          NaN                NaN        NaN   \n",
       "581        71.71               11.0        NaN   \n",
       "582          NaN                NaN        NaN   \n",
       "\n",
       "                                              warnings  \n",
       "0                                                  NaN  \n",
       "1                                                  NaN  \n",
       "2                                                  NaN  \n",
       "3                                                  NaN  \n",
       "4                                                  NaN  \n",
       "..                                                 ...  \n",
       "578  Genome not assigned to closest species as it f...  \n",
       "579                                                NaN  \n",
       "580                                                NaN  \n",
       "581                                                NaN  \n",
       "582                                                NaN  \n",
       "\n",
       "[583 rows x 62 columns]"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(binning\n",
    " .iloc[:, 1:]\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12f63b9b-6a11-418f-b5cb-d7670f6fcca7",
   "metadata": {},
   "source": [
    "### data from Genomebinning/Quast/quast_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "bfce9aee-87a7-4782-8379-3b5cc4932b8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "quast = pd.read_csv('quast_summary.tsv',\n",
    "                    sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "66e53ac9-0995-4c43-a352-4b18109eff5b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th># contigs (&gt;= 0 bp)</th>\n",
       "      <th># contigs (&gt;= 1000 bp)</th>\n",
       "      <th># contigs (&gt;= 5000 bp)</th>\n",
       "      <th># contigs (&gt;= 10000 bp)</th>\n",
       "      <th># contigs (&gt;= 25000 bp)</th>\n",
       "      <th># contigs (&gt;= 50000 bp)</th>\n",
       "      <th>Total length (&gt;= 0 bp)</th>\n",
       "      <th>Total length (&gt;= 1000 bp)</th>\n",
       "      <th>Total length (&gt;= 5000 bp)</th>\n",
       "      <th>...</th>\n",
       "      <th># contigs</th>\n",
       "      <th>Largest contig</th>\n",
       "      <th>Total length</th>\n",
       "      <th>GC (%)</th>\n",
       "      <th>N50</th>\n",
       "      <th>N75</th>\n",
       "      <th>L50</th>\n",
       "      <th>L75</th>\n",
       "      <th># N's per 100 kbp</th>\n",
       "      <th># predicted rRNA genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MEGAHIT-MetaBAT2-CAPES_S7.1.fa</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>201428</td>\n",
       "      <td>201428</td>\n",
       "      <td>201428</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>102222</td>\n",
       "      <td>201428</td>\n",
       "      <td>48.22</td>\n",
       "      <td>102222</td>\n",
       "      <td>68750</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MEGAHIT-MetaBAT2-CAPES_S7.11.fa</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>411934</td>\n",
       "      <td>411934</td>\n",
       "      <td>411934</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>183221</td>\n",
       "      <td>411934</td>\n",
       "      <td>44.58</td>\n",
       "      <td>109964</td>\n",
       "      <td>70155</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MEGAHIT-MetaBAT2-CAPES_S7.10.fa</td>\n",
       "      <td>243</td>\n",
       "      <td>243</td>\n",
       "      <td>127</td>\n",
       "      <td>53</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1728819</td>\n",
       "      <td>1728819</td>\n",
       "      <td>1377526</td>\n",
       "      <td>...</td>\n",
       "      <td>243</td>\n",
       "      <td>49968</td>\n",
       "      <td>1728819</td>\n",
       "      <td>39.43</td>\n",
       "      <td>10178</td>\n",
       "      <td>5643</td>\n",
       "      <td>53</td>\n",
       "      <td>112</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEGAHIT-MetaBAT2-CAPES_S7.13.fa</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>403950</td>\n",
       "      <td>403950</td>\n",
       "      <td>403950</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>143029</td>\n",
       "      <td>403950</td>\n",
       "      <td>43.63</td>\n",
       "      <td>100706</td>\n",
       "      <td>91934</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEGAHIT-MetaBAT2-CAPES_S7.12.fa</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>364182</td>\n",
       "      <td>364182</td>\n",
       "      <td>364182</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>225856</td>\n",
       "      <td>364182</td>\n",
       "      <td>57.26</td>\n",
       "      <td>225856</td>\n",
       "      <td>54475</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>SPAdesHybrid-MetaBAT2-CAPES_S11.55.fa</td>\n",
       "      <td>472</td>\n",
       "      <td>472</td>\n",
       "      <td>93</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1729859</td>\n",
       "      <td>1729859</td>\n",
       "      <td>690291</td>\n",
       "      <td>...</td>\n",
       "      <td>472</td>\n",
       "      <td>24414</td>\n",
       "      <td>1729859</td>\n",
       "      <td>59.52</td>\n",
       "      <td>4101</td>\n",
       "      <td>2734</td>\n",
       "      <td>132</td>\n",
       "      <td>261</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>SPAdesHybrid-MetaBAT2-CAPES_S11.59.fa</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>1019881</td>\n",
       "      <td>1019881</td>\n",
       "      <td>1019881</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>257271</td>\n",
       "      <td>1019881</td>\n",
       "      <td>44.75</td>\n",
       "      <td>110185</td>\n",
       "      <td>80799</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1023.55</td>\n",
       "      <td>1 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>SPAdesHybrid-MetaBAT2-CAPES_S11.57.fa</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>742576</td>\n",
       "      <td>742576</td>\n",
       "      <td>742576</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>235546</td>\n",
       "      <td>742576</td>\n",
       "      <td>42.00</td>\n",
       "      <td>187388</td>\n",
       "      <td>144928</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>472.41</td>\n",
       "      <td>1 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>SPAdesHybrid-MetaBAT2-CAPES_S11.58.fa</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>322085</td>\n",
       "      <td>322085</td>\n",
       "      <td>322085</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>123230</td>\n",
       "      <td>322085</td>\n",
       "      <td>41.58</td>\n",
       "      <td>95563</td>\n",
       "      <td>95374</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>SPAdesHybrid-MetaBAT2-CAPES_S11.60.fa</td>\n",
       "      <td>89</td>\n",
       "      <td>89</td>\n",
       "      <td>73</td>\n",
       "      <td>48</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>1269347</td>\n",
       "      <td>1269347</td>\n",
       "      <td>1220258</td>\n",
       "      <td>...</td>\n",
       "      <td>89</td>\n",
       "      <td>58771</td>\n",
       "      <td>1269347</td>\n",
       "      <td>33.05</td>\n",
       "      <td>20699</td>\n",
       "      <td>11911</td>\n",
       "      <td>20</td>\n",
       "      <td>39</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0 + 0 part</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>583 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Assembly  # contigs (>= 0 bp)  \\\n",
       "0           MEGAHIT-MetaBAT2-CAPES_S7.1.fa                    4   \n",
       "1          MEGAHIT-MetaBAT2-CAPES_S7.11.fa                    4   \n",
       "2          MEGAHIT-MetaBAT2-CAPES_S7.10.fa                  243   \n",
       "3          MEGAHIT-MetaBAT2-CAPES_S7.13.fa                    5   \n",
       "4          MEGAHIT-MetaBAT2-CAPES_S7.12.fa                    5   \n",
       "..                                     ...                  ...   \n",
       "578  SPAdesHybrid-MetaBAT2-CAPES_S11.55.fa                  472   \n",
       "579  SPAdesHybrid-MetaBAT2-CAPES_S11.59.fa                   12   \n",
       "580  SPAdesHybrid-MetaBAT2-CAPES_S11.57.fa                    6   \n",
       "581  SPAdesHybrid-MetaBAT2-CAPES_S11.58.fa                    4   \n",
       "582  SPAdesHybrid-MetaBAT2-CAPES_S11.60.fa                   89   \n",
       "\n",
       "     # contigs (>= 1000 bp)  # contigs (>= 5000 bp)  # contigs (>= 10000 bp)  \\\n",
       "0                         4                       4                        4   \n",
       "1                         4                       4                        4   \n",
       "2                       243                     127                       53   \n",
       "3                         5                       5                        5   \n",
       "4                         5                       5                        4   \n",
       "..                      ...                     ...                      ...   \n",
       "578                     472                      93                       10   \n",
       "579                      12                      12                       11   \n",
       "580                       6                       6                        6   \n",
       "581                       4                       4                        3   \n",
       "582                      89                      73                       48   \n",
       "\n",
       "     # contigs (>= 25000 bp)  # contigs (>= 50000 bp)  Total length (>= 0 bp)  \\\n",
       "0                          2                        2                  201428   \n",
       "1                          4                        3                  411934   \n",
       "2                          4                        0                 1728819   \n",
       "3                          4                        3                  403950   \n",
       "4                          4                        2                  364182   \n",
       "..                       ...                      ...                     ...   \n",
       "578                        0                        0                 1729859   \n",
       "579                       10                        8                 1019881   \n",
       "580                        5                        4                  742576   \n",
       "581                        3                        3                  322085   \n",
       "582                       15                        1                 1269347   \n",
       "\n",
       "     Total length (>= 1000 bp)  Total length (>= 5000 bp)  ...  # contigs  \\\n",
       "0                       201428                     201428  ...          4   \n",
       "1                       411934                     411934  ...          4   \n",
       "2                      1728819                    1377526  ...        243   \n",
       "3                       403950                     403950  ...          5   \n",
       "4                       364182                     364182  ...          5   \n",
       "..                         ...                        ...  ...        ...   \n",
       "578                    1729859                     690291  ...        472   \n",
       "579                    1019881                    1019881  ...         12   \n",
       "580                     742576                     742576  ...          6   \n",
       "581                     322085                     322085  ...          4   \n",
       "582                    1269347                    1220258  ...         89   \n",
       "\n",
       "     Largest contig  Total length  GC (%)     N50     N75  L50  L75  \\\n",
       "0            102222        201428   48.22  102222   68750    1    2   \n",
       "1            183221        411934   44.58  109964   70155    2    3   \n",
       "2             49968       1728819   39.43   10178    5643   53  112   \n",
       "3            143029        403950   43.63  100706   91934    2    3   \n",
       "4            225856        364182   57.26  225856   54475    1    2   \n",
       "..              ...           ...     ...     ...     ...  ...  ...   \n",
       "578           24414       1729859   59.52    4101    2734  132  261   \n",
       "579          257271       1019881   44.75  110185   80799    3    6   \n",
       "580          235546        742576   42.00  187388  144928    2    3   \n",
       "581          123230        322085   41.58   95563   95374    2    3   \n",
       "582           58771       1269347   33.05   20699   11911   20   39   \n",
       "\n",
       "     # N's per 100 kbp  # predicted rRNA genes  \n",
       "0                 0.00              0 + 0 part  \n",
       "1                 0.00              0 + 0 part  \n",
       "2                 0.00              0 + 0 part  \n",
       "3                 0.00              0 + 0 part  \n",
       "4                 0.00              0 + 0 part  \n",
       "..                 ...                     ...  \n",
       "578               0.00              0 + 0 part  \n",
       "579            1023.55              1 + 0 part  \n",
       "580             472.41              1 + 0 part  \n",
       "581               0.00              0 + 0 part  \n",
       "582               0.00              0 + 0 part  \n",
       "\n",
       "[583 rows x 23 columns]"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "766504b6-5df8-440b-bccf-b72ee689026c",
   "metadata": {},
   "outputs": [],
   "source": [
    "nextflow run nf-core/taxprofiler --input input.csv --databases databases.csv --outdir testtaxout -profile docker --run_kraken2 --run_kaiju\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
