# virusclass_nextflow logfile

## 2022-08-18
`To get BOWTIE2 process to run for every output tuple from FASTP, it was cruical to add the index as channel.value, to get the combinatorics!`

`bowtie`: The log file does not work with 2> .log... WHY?

#### Download kaiju virus db:
```bash
wget https://kaiju.binf.ku.dk/database/kaiju_db_viruses_2022-03-29.tgz
tar zxvf kaiju_db_viruses_2022-03-29.tgz 
```

The kaiju process works but produces a non file... Maybe because it is also from the stdout channel? 

Everything is nonefile!! WHY???
**SOLUTION TO ABOVE PROBLEM**
I had the argument cleanup = true in the .config file, which deletes content of work file. Since I had NOT put the publishDir mode as copy, the final results were just a symlink to the deleted work directory!. 

## 2022-08-19 
* Think I need to run kaiju against bigger database, containing both bacteria and viruses. 
`download the refseq database from kaijus server`
```bash
wget https://kaiju.binf.ku.dk/database/kaiju_db_refseq_2022-03-23.tgz
tar zxvf kaiju_db_refseq_2022-03-23.tgz
```

In `krona2table`the -u flag exists, which: *Unclassified reads are not counted for the total reads when calculating percentages for classified reads.*
Include or not include this? 

The KAIJU2TABLE module results in the error:
`unknown recognition error type: groovyjarjarantlr4.v4.runtime.LexerNoViableAltException` and complains that it does not have an input file. 

Trying to run in outside Nextflow:
```bash
kaiju2table -t /home/viller/virusclass/databases/kaiju_db/refseq/nodes.dmp -n /home/viller/virusclass/databases/kaiju_db/refseq/names.dmp -r genus -e -o kaiju_summary.tsv Dol1_S19_L001.kaiju.out
```
The above command works! Why does it not work inside nextflow main.nf?

**NOW the KAIJU2TABLE works, without me changing anything...** WEIRD!! 

* Download standard kraken database from https://benlangmead.github.io/aws-indexes/k2
```bash 
wget https://genome-idx.s3.amazonaws.com/kraken/k2_standard_20220607.tar.gz
```

Trying kraken2 command outside nextflow:
```bash
kraken2 \
    --db /home/viller/virusclass/databases/kraken/standard/ \
    --report testar.kraken2.report.tsv \
    --use-mpa-style \
    --use-names \
    --paired \
    /home/viller/virusclass/results/fastp/reads/Dol1_S19_L001_filt_R1.fastq.gz \
    /home/viller/virusclass/results/fastp/reads/Dol1_S19_L001_filt_R2.fastq.gz

# must unzip the krakendb tar first


```
Above command works! 

**Having problems with the megahit module. It will not save the file where it needs to go, so fiddeling around a little bit there. Otherwise it seems to work.**
FINALLY IT WORKS!
Had to do this: 
```bash
megahit \
    -1 ${reads[0]} \
    -2 ${reads[1]} \
    --out-dir megahit \
    --out-prefix ${sample_id}
    
    mv megahit/${sample_id}.contigs.fa ${sample_id}.contigs.fa
```

Added config_length.py to extract the length of the contigs generated by megahit. 
### **NB!** Had to put the .py script in the bin/ and put `#!/usr/bin/env python` in the python script!!!

**Will PROBABLY skip BUSCO, QUAST, prokka and prodigal.**
Will continue with CAT and gtdb and then... --> Phylogeny tree with mash!

### To install metabat2:
```bash
mamba install -c bioconda/label/cf201901 metabat2

```
Runned metabat2, with error:
```bash
Command error:
  [Error!] can't open input file Dol1_S19_L001_bins.table.tsv
  [Error!] There are no lines in the abundance depth file or fasta file!
```
**The bam file is generated like: align the raw reads to the assembled contigs** 
*"First we need to map the reads back against the assembly to get coverage information"*



Tutorial for binning here: https://www.youtube.com/watch?v=q9U0uTFRsl4&ab_channel=BioinformaticsVirtualCoordinationNetwork
*https://github.com/edgraham/BinSanity/wiki/Usage*
-- https://www.hadriengourle.com/tutorials/meta_assembly/

* Wrote python script for renaming fasta files to have headers like: contig_1, contig_2 etc...


### METABAT2 requires contigs > 2kb to be accurate. This is usually done by assembly of reads from many different samples from the same site. 

create CONTIGS2INDEX:
* map the reads back against the assembly to get coverage information 


## 2022-08-22 Monday

* Continue with bowtie2 alignment 
* samtools module
    * sam to bam
    * index and sort 
    
 
Which files should align to the new index of assembled contigs? The "unaligned" reads from the first bowtie2 step, or the files from FASTP? 
`Right now it aligned the host-read-filtered fastq files to the contigs index.`

**Need to output the folder name of the CONTIGS2INDEX process, not the files in it.** 
Error now:
```bash
Error executing process > 'BOWTIE2_ALIGN2CONTIGS (1)'

Caused by:  Process `BOWTIE2_ALIGN2CONTIGS (1)` terminated with an error exit status (255)Command executed:
  bowtie2     -x [Dol1_S19_L001, [/home/viller/virusclass/virusclassification_nextflow/work/71/a25811cb79674320e38718cd843fed/Dol1_S19_L001.1.bt2, /home/viller/virusclass/virusclassification_nextflow/work/71/a25811cb79674320e38718cd843fed/Dol1_S19_L001.2.bt2, /home/viller/virusclass/virusclassification_nextflow/work/71/a25811cb79674320e38718cd843fed/Dol1_S19_L001.3.bt2, /home/viller/virusclass/virusclassification_nextflow/work/71/a25811cb79674320e38718cd843fed/Dol1_S19_L001.4.bt2, /home/viller/virusclass/virusclassification_nextflow/work/71/a25811cb79674320e38718cd843fed/Dol1_S19_L001.rev.1.bt2, /home/viller/virusclass/virusclassification_nextflow/work/71/a25811cb79674320e38718cd843fed/Dol1_S19_L001.rev.2.bt2]]     -1 Dol1_S19_L001_unaligned.1.fastq     -2 Dol1_S19_L001_unaligned.2.fastq     -S Dol1_S19_L001_aligned.sam     2> Dol1_S19_L001.bowtie.log

Command exit status:
  255
```
The output from `CONTIGS2INDEX` **CANNOT** be a tuple! 
Change from
```
output:
    tuple val(sample_id), path('*.bt2'), emit: index
    
# to:
output:
    val(sample_id), emit: sample_id
    path('*.bt2'), emit: index
```

Trying to use the operator: **getParent** to get the folder of the contig index. 
Documentation: https://buildmedia.readthedocs.org/media/pdf/nextflow1/latest/nextflow1.pdf


changing from:
```
path('*.bt2'), emit: index
# to
path('${sample_id}'), emit: index
```
The above did not work, it must be: `path('*.bt2'), emit: index`. Therefore I need to get the parentfolder with operators like: operator: **getParent** 
in the main.nf script. 


Trying many different things related to all of the above, but it in the end it was hard to get it to work. The solution below, however, worked:
```nextflow
process TESTAR {  
    echo true

    input:
    tuple val(sample_id), path(reads)
    
    
    script:
    index = params.bowtie2_align2contigs.toString() + '/' + sample_id
    """
    echo ${index}
   
    """   
}
```

In CONTIS2INDEX:
```bash
# instead of:
output:
    val(sample_id), emit: sample_id
    val('*.bt2'), emit: index
# change to:
```

Testing bowtie2 from command line:
```bash
bowtie2 -x /home/viller/virusclass/databases/bowtie2/Dol1_S19_L001/Dol1_S19_L001 -1 /home/viller/virusclass/results/bowtie2/unaligned/Dol1_S19_L001_unaligned.1.fastq -2 /home/viller/virusclass/results/bowtie2/unaligned/Dol1_S19_L001_unaligned.2.fastq -S TESTARBOWTIE_aligned.sam
```
The above works...

### Now the ALIGN2CONTIGS works
Had to publishdir the new index of the contigs to a absolute path and use: 
`index = params.bowtie2_contigs_index.toString() + '/' + sample_id + '/' + sample_id`
To get it to work. 

install samtools:
```bash
mamba install samtools==1.11
```

**YES!!** Finally fixed the bowtieALIGN2CONTIGS problem (that it could not find the index). Fixed it with this line:
```bash
index = index_files[0].toRealPath().toString().split('\\.')[0]
```

