{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f43d6bb0-d607-436e-b04a-f2e7ade5d6a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "def wrangle_kaiju_megahit_cat(kaiju: str, \n",
    "                              cat: str, \n",
    "                              megahit: str, \n",
    "                              name: str) -> None:\n",
    "    '''\n",
    "    Wrangles and merges the CAT, kaiju and MEGAHIT output from the MEGAHIT contigs.\n",
    "    :param str kaiju: The kaiju out name path\n",
    "    :param str cat: The CAT out txt path\n",
    "    :param str megahit: The path to the megahit csv file. \n",
    "    :param str name: Name of the new csv file. \n",
    "    :returns: New merged csv file\n",
    "    '''\n",
    "    \n",
    "    kaiju_raw = pd.read_csv(kaiju,\n",
    "                    sep='\\t',\n",
    "                    header=None,\n",
    "                    usecols=[1, 2, 6, 7],\n",
    "                    names=['name', 'taxon_id', 'aa_match', 'taxonomy'])\n",
    "    \n",
    "    megahit = pd.read_csv(megahit)\n",
    "\n",
    "    kaiju = (kaiju_raw.merge(megahit, on='name', how='outer')\n",
    "     .sort_values('length', ascending=False)\n",
    "     #.dropna()\n",
    "     .assign(taxonomy=lambda x: x.taxonomy.str.split(';').str[:-1])\n",
    "     .assign(last_level_kaiju=lambda x: x.taxonomy.str[-1])\n",
    "     .assign(second_level_kaiju=lambda x: x.taxonomy.str[-2])\n",
    "     .assign(third_level_kaiju=lambda x: x.taxonomy.str[-3])\n",
    "     .assign(kingdom_kaiju=lambda x: np.select([x.taxonomy.str[0] != 'cellular organisms'],\n",
    "                                                   [x.taxonomy.str[0]],\n",
    "                                                   default=x.taxonomy.str[1])))\n",
    "    \n",
    "    \n",
    "    cat_raw = pd.read_csv(cat,\n",
    "                  sep='\\t')\n",
    "\n",
    "    cat = (\n",
    "        cat_raw\n",
    "     .rename(columns={'# contig': 'name',\n",
    "                      'species': 'last_level_cat',\n",
    "                      'genus': 'second_level_cat',\n",
    "                      'family': 'third_level_cat'})\n",
    "    # .loc[lambda x: x.classification != 'no taxid assigned']\n",
    "    # .loc[lambda x: x['superkingdom'] != 'no support']\n",
    "    # .loc[lambda x: x['phylum'] != 'no support']\n",
    "     .fillna(\"\")\n",
    "     .drop(columns=['lineage', 'lineage scores'])\n",
    "     .assign(kingdom_cat=lambda x: x['superkingdom'].str[:-6])\n",
    "     .drop(columns='superkingdom')\n",
    "    )\n",
    "\n",
    "    merged = (kaiju.merge(cat, on='name', how='outer')\n",
    "              .sort_values('length', ascending=False))\n",
    "    \n",
    "    return merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "39681b6d-97c3-4a77-864a-ec04f2bab08e",
   "metadata": {},
   "outputs": [],
   "source": [
    "kaiju = \"~/clinical-genomics/pandemic-preparedness/virusclassification_nextflow/results/sample_01_S8/kaiju/megahit/sample_01_S8_table_megahit.tsv\"\n",
    "cat = \"~/clinical-genomics/pandemic-preparedness/virusclassification_nextflow/results/sample_01_S8/cat/CAT_sample_01_S8_contigs_names.txt\"\n",
    "megahit = \"~/clinical-genomics/pandemic-preparedness/virusclassification_nextflow/results/sample_01_S8/megahit/sample_01_S8.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "92af98c8-88d3-4a2d-b1c7-96f2b2960a47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># contig</th>\n",
       "      <th>classification</th>\n",
       "      <th>reason</th>\n",
       "      <th>lineage</th>\n",
       "      <th>lineage scores</th>\n",
       "      <th>superkingdom</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>k141_0</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no ORFs found</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>k141_1</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no ORFs found</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>k141_2</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no hits to database</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>k141_3</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no ORFs found</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>k141_4</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no hits to database</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # contig     classification               reason lineage lineage scores  \\\n",
       "0   k141_0  no taxid assigned        no ORFs found                          \n",
       "1   k141_1  no taxid assigned        no ORFs found                          \n",
       "2   k141_2  no taxid assigned  no hits to database                          \n",
       "3   k141_3  no taxid assigned        no ORFs found                          \n",
       "4   k141_4  no taxid assigned  no hits to database                          \n",
       "\n",
       "  superkingdom phylum class order family genus species  \n",
       "0                                                       \n",
       "1                                                       \n",
       "2                                                       \n",
       "3                                                       \n",
       "4                                                       "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cat_df = pd.read_csv(cat, sep=\"\\t\").fillna(\" \")\n",
    "\n",
    "cat_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "38b9f64b-61fe-4fd4-9352-6df2d5e22f97",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2220124/3605304518.py:16: FutureWarning: Defining usecols with out of bounds indices is deprecated and will raise a ParserError in a future version.\n",
      "  kaiju_raw = pd.read_csv(kaiju,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>aa_match</th>\n",
       "      <th>taxonomy</th>\n",
       "      <th>length</th>\n",
       "      <th>sequence</th>\n",
       "      <th>last_level_kaiju</th>\n",
       "      <th>second_level_kaiju</th>\n",
       "      <th>third_level_kaiju</th>\n",
       "      <th>kingdom_kaiju</th>\n",
       "      <th>classification</th>\n",
       "      <th>reason</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>third_level_cat</th>\n",
       "      <th>second_level_cat</th>\n",
       "      <th>last_level_cat</th>\n",
       "      <th>kingdom_cat</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>k141_0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>498.0</td>\n",
       "      <td>GTCGCCTCTACATATAAATCTTTCAACAATTGCTGCATAGAAGGGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no ORFs found</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>k141_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>351.0</td>\n",
       "      <td>TGATATTTAACCTTTGGTCTGTTTAATTTGCTGTCCTATTTTAAGT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no ORFs found</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>k141_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>315.0</td>\n",
       "      <td>TAAGATAACATGCTTAACCTTTTAAAGGCATGCTGCCATTCCCAAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no hits to database</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>k141_4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>264.0</td>\n",
       "      <td>AGGGTCATTGTGTGAAGCCCAGGCTGAGTGTGTCTGTCTCTTACAC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no hits to database</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>k141_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>256.0</td>\n",
       "      <td>ACTTCTTACATATATTGATTAACACCTCATGTCTCTCACCTGCCTG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no taxid assigned</td>\n",
       "      <td>no ORFs found</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>percent</td>\n",
       "      <td>reads</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>100.000000</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         name taxon_id aa_match taxonomy  length  \\\n",
       "0      k141_0      NaN      NaN      NaN   498.0   \n",
       "1      k141_1      NaN      NaN      NaN   351.0   \n",
       "2      k141_2      NaN      NaN      NaN   315.0   \n",
       "3      k141_4      NaN      NaN      NaN   264.0   \n",
       "4      k141_3      NaN      NaN      NaN   256.0   \n",
       "5     percent    reads      NaN      NaN     NaN   \n",
       "6    0.000000        0      NaN      NaN     NaN   \n",
       "7  100.000000        5      NaN      NaN     NaN   \n",
       "\n",
       "                                            sequence last_level_kaiju  \\\n",
       "0  GTCGCCTCTACATATAAATCTTTCAACAATTGCTGCATAGAAGGGT...              NaN   \n",
       "1  TGATATTTAACCTTTGGTCTGTTTAATTTGCTGTCCTATTTTAAGT...              NaN   \n",
       "2  TAAGATAACATGCTTAACCTTTTAAAGGCATGCTGCCATTCCCAAA...              NaN   \n",
       "3  AGGGTCATTGTGTGAAGCCCAGGCTGAGTGTGTCTGTCTCTTACAC...              NaN   \n",
       "4  ACTTCTTACATATATTGATTAACACCTCATGTCTCTCACCTGCCTG...              NaN   \n",
       "5                                                NaN              NaN   \n",
       "6                                                NaN              NaN   \n",
       "7                                                NaN              NaN   \n",
       "\n",
       "  second_level_kaiju third_level_kaiju kingdom_kaiju     classification  \\\n",
       "0                NaN               NaN           NaN  no taxid assigned   \n",
       "1                NaN               NaN           NaN  no taxid assigned   \n",
       "2                NaN               NaN           NaN  no taxid assigned   \n",
       "3                NaN               NaN           NaN  no taxid assigned   \n",
       "4                NaN               NaN           NaN  no taxid assigned   \n",
       "5                NaN               NaN           NaN                NaN   \n",
       "6                NaN               NaN           NaN                NaN   \n",
       "7                NaN               NaN           NaN                NaN   \n",
       "\n",
       "                reason phylum class order third_level_cat second_level_cat  \\\n",
       "0        no ORFs found                                                       \n",
       "1        no ORFs found                                                       \n",
       "2  no hits to database                                                       \n",
       "3  no hits to database                                                       \n",
       "4        no ORFs found                                                       \n",
       "5                  NaN    NaN   NaN   NaN             NaN              NaN   \n",
       "6                  NaN    NaN   NaN   NaN             NaN              NaN   \n",
       "7                  NaN    NaN   NaN   NaN             NaN              NaN   \n",
       "\n",
       "  last_level_cat kingdom_cat  \n",
       "0                             \n",
       "1                             \n",
       "2                             \n",
       "3                             \n",
       "4                             \n",
       "5            NaN         NaN  \n",
       "6            NaN         NaN  \n",
       "7            NaN         NaN  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wrangle_kaiju_megahit_cat(kaiju, cat, megahit, name=\"hej\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
